---
title: "SGG - How do CPR taxa do?"
author: "Andr√© Soares"
date: "8 January 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r initial loadings, include=FALSE}
library(dada2); packageVersion("dada2")
library(ggplot2); packageVersion("ggplot2")
library(phyloseq); packageVersion("phyloseq")
library(data.table); packageVersion("data.table")
library(DESeq2); packageVersion("DESeq2")
library(dplyr)
library(plyr)
library(magrittr)
library(RColorBrewer)
library(phangorn)
library(DECIPHER)
library("structSSI")
library(ips)
library(msa)
library(picante)
library(igraph)
library(cowplot)
library(plotly)
library(lemon)
library(ggcorrplot)
library(GGally)
```

```{r load SV data, echo=FALSE}
SGG_SVt = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/seqtab_final.rds")
SGG_Tax = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/tax_final.rds")
sgg_metadata<-read.csv("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_metadata/SGG_metadata.csv", header=TRUE)
order<-rownames(SGG_SVt)
sgg_metadata_ord<-sgg_metadata[match(order, sgg_metadata$Sample_Code),]
rownames(sgg_metadata_ord) <- sgg_metadata_ord$Sample_Code
ps <- phyloseq(otu_table(SGG_SVt, taxa_are_rows=FALSE),
               sample_data(sgg_metadata_ord), 
               tax_table(SGG_Tax))
ps.a = subset_samples(ps, Site.name != "Taffs Well PUMPED")
ps.b = subset_samples(ps.a, Sample_type != "Control")
ps.b <- prune_taxa(taxa_sums(ps.b) > 0, ps.b)
#order months
sample_data(ps.b)$Month = factor(sample_data(ps.b)$Month, 
                                 levels = c("April","August","December"))
#to relative abundances
ps.b = subset_taxa(ps.b, Kingdom %in% c("Archaea", "Bacteria"))
ps.b.r <-  transform_sample_counts(ps.b, function(x) {x/sum(x)} ) 
```

##1 - Main CPR phyla across the dataset

So, as we know from before, although Proteobacteria dominate this dataset, a significant portion of all profiles is taken up by either Omnitrophica or Parcubacteria. Also, in general, these make up quite the proportion of all classified SVs:

```{r omni parcu svs}
sv.phyl.p<-as.data.frame(table(tax_table(ps.b.r)[, "Phylum"], exclude = NULL))
sv.phyl.p.ord <- sv.phyl.p[order(-sv.phyl.p$Freq),] 
head(sv.phyl.p.ord, n=10)
(10064+6314)/sum(sv.phyl.p.ord$Freq)
```

So around 23% of the dataset is CPR. But given how novel these phyla are, classification is still lacking at deeper taxonomic levels.

```{r omniparcu uncs}
ps.b.r.cpr = subset_taxa(ps.b.r, Phylum %in% c("Omnitrophica", "Parcubacteria"))
ps.b.r.cpr.c<-as.data.frame(table(tax_table(ps.b.r.cpr)[, "Class"], exclude = NULL))
ps.b.r.cpr.c.ord <- ps.b.r.cpr.c[order(-ps.b.r.cpr.c$Freq),] 
head(ps.b.r.cpr.c.ord, n=10)
11198/sum(ps.b.r.cpr.c.ord$Freq)
```

Even at class level, ~68% of Omnitrophica and Parcubacteria SVs are unclassified. Let's take a closer look at their distribution though.

```{r general cpr taxonomy}
ps.b.r.cpr.glom <- tax_glom(ps.b.r.cpr, taxrank = 'Phylum', NArm = FALSE)
ps.b.r.cpr.glom.psdf <- data.table(psmelt(ps.b.r.cpr.glom))
ps.b.r.cpr.glom.psdf$Phylum <- as.character(ps.b.r.cpr.glom.psdf$Phylum)
ps.b.r.cpr.glom.psdf.plot<-ggplot(ps.b.r.cpr.glom.psdf[Abundance > 0], 
                                    aes(x = Site.name, y = Abundance/3, fill = Phylum)) + 
  facet_grid(Month~.) +
  geom_bar(stat = "identity") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1)) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance\n")
```
```{r , fig.height=6, fig.width=8, out.width = '500%'}
ps.b.r.cpr.glom.psdf.plot
```

Are these related to time of sampling for example? CPR phyla have been described to mainly be composed of ultrasmall bacteria frequently posed to be symbiotic to larger bacteria. Is there evidence for this in this dataset?

```{r cpr month}
ps.b.r.cpr.glom.temp <- tax_glom(ps.b.r.cpr, taxrank = 'Class', NArm = FALSE)
ps.b.r.cpr.glom.psdf.temp <- data.table(psmelt(ps.b.r.cpr.glom.temp))
ps.b.r.cpr.glom.psdf.temp$Class<-as.character(ps.b.r.cpr.glom.psdf.temp$Class)
ps.b.r.cpr.glom.psdf.temp <- data.frame(lapply(ps.b.r.cpr.glom.psdf.temp, 
                                               function(x) {gsub("Candidatus_", "Cand. ", x)}))
levels(ps.b.r.cpr.glom.psdf.temp$Class)<-c(levels(ps.b.r.cpr.glom.psdf.temp$Class),c("Unc. Omnitrophica","Unc. Parcubacteria"))
ps.b.r.cpr.glom.psdf.temp$Class[is.na(ps.b.r.cpr.glom.psdf.temp$Class) & ps.b.r.cpr.glom.psdf.temp$Phylum=="Omnitrophica"] <- "Unc. Omnitrophica"
ps.b.r.cpr.glom.psdf.temp$Class[is.na(ps.b.r.cpr.glom.psdf.temp$Class) & ps.b.r.cpr.glom.psdf.temp$Phylum=="Parcubacteria"] <- "Unc. Parcubacteria"
ps.b.r.cpr.glom.psdf.temp$Class<-as.character(ps.b.r.cpr.glom.psdf.temp$Class)
ps.b.r.cpr.glom.psdf.temp$Abundance<-as.numeric(as.character(ps.b.r.cpr.glom.psdf.temp$Abundance))
ps.b.r.cpr.glom.psdf.temp$Month = factor(ps.b.r.cpr.glom.psdf.temp$Month,
                                           levels=c("December", "August", "April"))
ps.b.r.cpr.glom.psdf.temp.f<-subset(ps.b.r.cpr.glom.psdf.temp,
                                                  Class %in% c("Unc. Omnitrophica", "Unc. Parcubacteria", "Cand. Azambacteria",
                                                               "Cand. Jorgensenbacteria","Cand. Nomurabacteria","Cand. Moranbacteria"))
ps.b.r.cpr.glom.psdf.temp.f$Class = factor(ps.b.r.cpr.glom.psdf.temp.f$Class,
                                           levels=c("Unc. Omnitrophica", "Unc. Parcubacteria", "Cand. Azambacteria",
                                                    "Cand. Jorgensenbacteria","Cand. Nomurabacteria","Cand. Moranbacteria"))
ps.b.r.cpr.glom.psdf.temp.f.plot <- ggplot(ps.b.r.cpr.glom.psdf.temp.f, 
       aes(x=Month, y=Abundance, color=Site.name)) +
  geom_point(aes(fill=Site.name, 
                 color=Site.name)) +
  geom_smooth(aes(group=Site.name),
              method="loess", se = FALSE) +
  facet_wrap(~Class, ncol=4)+
  theme(axis.title.y = element_blank(),
        axis.text.x = element_text(hjust=1, vjust=1, angle=25, size=8),
        axis.text.y = element_text(size=8),
        text = element_text(size=8)) +
  coord_flip() +
  scale_x_discrete(expand=c(0.05, 0.05)) +
  scale_color_manual(values = colorRampPalette(brewer.pal(12, "Paired"))(13))
```
```{r message=FALSE, warning=FALSE, , out.width = '100%'}
ps.b.r.cpr.glom.psdf.temp.f.plot
```

So, quite interesting stuff! Phylum Omnitrophica and member of the Parcubacteria phylum (now superphylum) Falkowbacteria have been implicated as being involved in dissimilatory sulfur cycling. The first seems to reduce sulfite via the siroheme-dependent anaerobic sulfite reductase (asr), while the latter hasn't had its sulfur cycling mechanisms completely unfolded. Both can use H, fatty acid metabolism-derived and organic C compounds as electron donors.

So, what next then? How about aligning all CPR sequences?

```{r cpr alignment}
seqs <- taxa_names(ps.b.r.cpr)
names(seqs) <- seqs
# alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA)
# writeXStringSet(alignment, filepath="/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/phyloseq_DataAnalysis/cpr_alignments.fasta")
cpr_alignment<-readDNAStringSet("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/phyloseq_DataAnalysis/cpr_alignments.fasta")
# identical(alignment, cpr_alignment)
# tree was generated within linux with
# FastTree -gtr -nt < cpr_alignments.fasta > cpr_alignments_tree
#using Fasttree v2.1.3, 
library(ape)
tree_Q = read.tree("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/phyloseq_DataAnalysis/cpr_alignments_tree")
is.rooted(tree_Q)
##randomly root your tree -- or you could put an outgroup in and then trim it after tree is rooted
tree_Q = root(tree_Q, 1, resolve.root = T)
ps.b.r.cpr.tree<-merge_phyloseq(ps.b.r.cpr, tree_Q)
ps.b.r.cpr.tree

Top50CPR <- names(sort(taxa_sums(ps.b.r.cpr.tree), TRUE)[1:50])
Top50CPR.ps.b.r.cpr.tree   <- prune_taxa(Top50CPR, ps.b.r.cpr.tree)
plot_tree(Top50CPR.ps.b.r.cpr.tree, ladderize=TRUE, 
          color="Class", shape="Phylum")
```

Ok this is getting weird - Wolfebacteria supposedly within Omnitrophica? Get a root! Let's do something useful!

