---
title: "SGG - What explains proteobacterial co-dynamics in the SGG dataset?"
author: "Andr√© Soares"
date: "23 March 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r initial loadings, message=FALSE, warning=FALSE, include=FALSE}
library(dada2); packageVersion("dada2")
library(ggplot2); packageVersion("ggplot2")
library(phyloseq); packageVersion("phyloseq")
library(data.table); packageVersion("data.table")
library(DESeq2); packageVersion("DESeq2")
library(dplyr)
library(plyr)
library(magrittr)
library(RColorBrewer)
library(phangorn)
library(DECIPHER)
library("structSSI")
library(ips)
library(msa)
library(picante)
library(igraph)
library(cowplot)
library(plotly)
library(lemon)
library(GGally)
library(randomcoloR)
library(knitr)
library(kableExtra)
library(ggpmisc)
library(ggpubr)
```

```{r load SV data, echo=FALSE}
SGG_SVt = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/seqtab_final.rds")
SGG_Tax = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/tax_final.rds")
sgg_metadata<-read.csv("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_metadata/SGG_metadata.csv", header=TRUE)
order<-rownames(SGG_SVt)
sgg_metadata_ord<-sgg_metadata[match(order, sgg_metadata$Sample_Code),]
rownames(sgg_metadata_ord) <- sgg_metadata_ord$Sample_Code
ps <- phyloseq(otu_table(SGG_SVt, taxa_are_rows=FALSE),
               sample_data(sgg_metadata_ord), 
               tax_table(SGG_Tax))
ps.a = subset_samples(ps, Site.name != "Taffs Well PUMPED")
ps.b = subset_samples(ps.a, Sample_type != "Control")
ps.b <- prune_taxa(taxa_sums(ps.b) > 0, ps.b)
#order months
sample_data(ps.b)$Month = factor(sample_data(ps.b)$Month, 
                                 levels = c("April","August","December"))
#to relative abundances
ps.b = subset_taxa(ps.b, Kingdom %in% c("Archaea", "Bacteria"))
ps.b.r <-  transform_sample_counts(ps.b, function(x) {x/sum(x)} ) 
```

##**1 -** High-level taxonomic view of the dataset:

```{r general proteobacterial taxonomy, echo=FALSE}
# ps.b.r.glom <- tax_glom(ps.b.r, taxrank = 'Phylum', NArm = FALSE)
# saveRDS(ps.b.r.glom, "/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/ps.b.r.glom.rds")
ps.b.r.glom = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/ps.b.r.glom.rds")
ps.b.r.glom.psdf <- data.table(psmelt(ps.b.r.glom))
ps.b.r.glom.psdf$Phylum <- as.character(ps.b.r.glom.psdf$Phylum)

ps.b.r.glom.psdf[, median := median(Abundance, na.rm = TRUE), 
                 by = "Phylum"]
ps.b.r.glom.psdf[(median <= 0.01), Phylum := "Other"]
#Removing Proteobacteria from previous table
ps.b.r.glom.psdf.noProteo<-ps.b.r.glom.psdf[!grepl("Proteobacteria", ps.b.r.glom.psdf$Phylum),]
#New table, with Proteobacterial classes only
ps.b.r.ProtCl = subset_taxa(ps.b.r, Phylum == "Proteobacteria")
ps.b.r.ProtCl.glom <- tax_glom(ps.b.r.ProtCl, taxrank = 'Class', NArm = FALSE)
ps.b.r.ProtCl.glom.psdf <- data.table(psmelt(ps.b.r.ProtCl.glom))
ps.b.r.ProtCl.glom.psdf$Class <- as.character(ps.b.r.ProtCl.glom.psdf$Class)

ps.b.r.ProtCl.glom.psdf[, median := median(Abundance, na.rm = TRUE), 
                        by = "Class"]
ps.b.r.ProtCl.glom.psdf[(median <= 0.01), Class := "Other Proteobacteria"]
ps.b.r.ProtCl.glom.psdf.noP <- subset(ps.b.r.ProtCl.glom.psdf, select = -Phylum)

names(ps.b.r.glom.psdf.noProteo)[names(ps.b.r.glom.psdf.noProteo) == 'Phylum'] <- 'Taxa'
names(ps.b.r.ProtCl.glom.psdf.noP)[names(ps.b.r.ProtCl.glom.psdf.noP) == 'Class'] <- 'Taxa'
ps.b.r.ProteoCl.AllPhy <- rbind(ps.b.r.glom.psdf.noProteo, ps.b.r.ProtCl.glom.psdf.noP)

ps.b.r.ProteoCl.AllPhy$Site.name = factor(ps.b.r.ProteoCl.AllPhy$Site.name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
ps.b.r.ProteoCl.AllPhy$Taxa = factor(ps.b.r.ProteoCl.AllPhy$Taxa, 
                                                       levels = c("Betaproteobacteria","Deltaproteobacteria",
                                                                  "Epsilonproteobacteria","Gammaproteobacteria","Other Proteobacteria",
                                                                  "Actinobacteria","Nitrospirae","Omnitrophica","Parcubacteria",
                                                                  "Other"))
ps.b.r.ProteoCl.AllPhy.plot<-ggplot(ps.b.r.ProteoCl.AllPhy[Abundance > 0], 
                                    aes(x = Site.name, y = Abundance/3, fill = Taxa)) + 
  facet_grid(Month~.) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = brewer.pal(10, "Paired"),
                     na.value = "gray40") +
  scale_y_continuous(labels = scales::percent) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 30, hjust = 1)) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance\n")
```
```{r echo=FALSE, fig.height=14, fig.width=20}
ps.b.r.ProteoCl.AllPhy.plot
```

At a first glance, a dataset dominated by Epsilonproteobacteria and Betaproteobacteria, with some CPR phyla (Omnitrophica and Parcubacteria) presence. Sites that smelt like sulphur during sampling are populated by Epsilonproteobacteria, while iron and iron-hydroxide filled sites have strong Betaproteobacterial presence on most occasions, showing Omnitrophica and Parcubacteria occupying close to 30-40% of the taxonomic profiles. Lindsay stands alone, showing CPR phyla occupying most of the profile and low-abundance but numerous phyla representing close to 40% of the profile throughout the year.

But is this seen at lower taxonomic levels?

##**2 -** Genus-level view of the dataset:

```{r genera plot, echo=FALSE}
ps.b.deep_128 = subset_taxa(ps.b.r, 
                            Family=="Gallionellaceae" | Family=="Helicobacteraceae" | Family=="Pseudomonadaceae" | Family=="Thiotrichaceae" | Family=="Hydrogenophilaceae" | Phylum=="Omnitrophica" | Phylum=="Parcubacteria")
ps.b.deep_128.glom <- tax_glom(ps.b.deep_128, taxrank = 'Genus', NArm = FALSE)
ps.b.deep_128.glom.df <- data.table(psmelt(ps.b.deep_128.glom))
levels(ps.b.deep_128.glom.df$Genus)<-c(levels(ps.b.deep_128.glom.df$Genus),
                                     c("Unc. Gallionellaceae","Unc. Helicobacteraceae", 
                                       "Omnitrophica","Parcubacteria","Unc. Omnitrophica","Unc. Parcubacteria",
                                       "Unc. Pseudomonadaceae","Unc. Thiotrichaceae","Unc. Hydrogenophilaceae"))
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Gallionellaceae"] <- "Unc. Gallionellaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Helicobacteraceae"] <- "Unc. Helicobacteraceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Pseudomonadaceae"] <- "Unc. Pseudomonadaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Thiotrichaceae"] <- "Unc. Thiotrichaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Hydrogenophilaceae"] <- "Unc. Hydrogenophilaceae"
ps.b.deep_128.glom.df$Genus[ps.b.deep_128.glom.df$Phylum=="Omnitrophica"] <- "Omnitrophica"
ps.b.deep_128.glom.df$Genus[ps.b.deep_128.glom.df$Phylum=="Parcubacteria"] <- "Parcubacteria"
ps.b.deep_128.glom.df$Site.name = factor(ps.b.deep_128.glom.df$Site.name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
# ps.b.r.glom.psdf.unk.glom <- tax_glom(ps.b.r, taxrank = 'Genus', NArm = FALSE)
# saveRDS(ps.b.r.glom.psdf.unk.glom, "/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/ps.b.r.glom.psdf.unk.glom.rds")
ps.b.r.glom.psdf.unk.glom = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_R/ps.b.r.glom.psdf.unk.glom.rds")
ps.b.r.glom.psdf.unk <- data.table(psmelt(ps.b.r.glom.psdf.unk.glom))
levels(ps.b.r.glom.psdf.unk$Phylum)<-c(levels(ps.b.r.glom.psdf.unk$Phylum),
                                     c("Unknown Phyla"))
levels(ps.b.r.glom.psdf.unk$Genus)<-c(levels(ps.b.r.glom.psdf.unk$Genus),
                                     c("Unknown Phyla"))
ps.b.r.glom.psdf.unk$Phylum[is.na(ps.b.r.glom.psdf.unk$Phylum)]="Unknown Phyla"
ps.b.r.glom.psdf.unk$Genus[ ps.b.r.glom.psdf.unk$Phylum=="Unknown Phyla"] <- "Unknown Phyla"
ps.b.r.glom.psdf.unkonly<-subset(ps.b.r.glom.psdf.unk, 
                                                  Phylum %in% c("Unknown Phyla"))
ps.b.deep_128.glom.df.unkonly <- rbind(ps.b.deep_128.glom.df, ps.b.r.glom.psdf.unkonly)

gen_cols = colorRampPalette(brewer.pal(8, "Accent"))(15)
gen_cols_c = gsub("#DD2456", "#9c183c", gen_cols)
gen_cols_c = gsub("#A0BAAC", "#749a85", gen_cols_c)
gen_cols_c = gsub("#E4B9A3", "#489fca", gen_cols_c)
gen_cols_c = gsub("#FEEB93", "#f58e03", gen_cols_c)
gen_cols_c = gsub("#FDC988", "#c988fd", gen_cols_c)
gen_cols_c = gsub("#D1DD9E", "#b6c965", gen_cols_c)
gen_cols_c = gsub("#FDC086", "#fc993b", gen_cols_c)
gen_cols_c = gsub("#FFFF99", "springgreen4", gen_cols_c)
gen_cols_c = gsub("#BEAED4", "violet", gen_cols_c)

ps.b.deep_128.glom.df.unkonly.f<-subset(ps.b.deep_128.glom.df.unkonly, 
                                                  Abundance > 0.03)
ps.b.deep_128.glom.df.unkonly.f$Genus = factor(ps.b.deep_128.glom.df.unkonly.f$Genus, 
                                                       levels = c("Gallionella","Sideroxydans","Ferriphaselus","Unc. Gallionellaceae",
                                                                  "Sulfurovum","Sulfurimonas","Sulfuricurvum","Sulfuricella","Sulfuriferula",
                                                                  "Pseudomonas","Thiothrix","Unc. Hydrogenophilaceae",
                                                                  "Parcubacteria","Omnitrophica",
                                                                  "Unknown Phyla"))
ps.b.deep_128.glom.df.plot <- ggplot(ps.b.deep_128.glom.df.unkonly.f, 
                                     aes(x = Genus, y = Abundance, fill = Genus)) + 
  facet_grid(Site.name~Month)+
  geom_hline(yintercept = 0.5,
           color="white",
           linetype="solid") +
  geom_point(aes(color=Genus, fill=Genus,
                 size=Abundance), alpha = 0.8) +
  scale_y_continuous(breaks = c(0.25,0.75),
                     limits = c(0,1) ) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.text.y = element_text(angle = 0),
        strip.placement = "outside",
        strip.background = element_rect(colour = "gray88",
                                        fill = "gray88"),
        panel.background = element_rect(colour = "gray91",
                                       fill = "gray91"),
        panel.grid.major.y = element_line(color = "white"),
        legend.title = element_text(face = "bold")) + 
  guides(fill = guide_legend(nrow = 16),
         shape = NULL,
         size = NULL,
         alpha = NULL) +
  ylab("Relative Abundance\n") +
  scale_fill_manual(values = gen_cols_c,
                    na.value = "gray48") +
  scale_color_manual(values = gen_cols_c,
                    na.value = "gray48")
```
```{r echo=FALSE, fig.width=14, message=FALSE, warning=FALSE, fig.height=11}
ps.b.deep_128.glom.df.plot
```

It seems that things are not that simple in this dataset! To put it simply, Epsilonproteobacteria-rich samples are co-dominated by Sulfuricurvum and Sulfurovum and Betaproteobacteria-rich samples by Gallionella and Sideroxydans. Lindsay, as seen before, is mostly made up of low-abundance taxa (not shown) with CPR phyla representing close to 50% of the profile across the year. Taff's well, albeit iron-rich, probably due to its external influences being on the edge of the coalfield, is populated by a mix of iron- and sulfur-oxidisers in April, which then turns into a FeOB-dominated profile in August, with Ferriphaselus going up to 50% in December.

It's unclear why bacteria with very similar metabolisms (iron- and sulfur-oxidisers) would co-occur and seemingly compete for dominance of the profiles this way. What could drive this? What do available genomes for Sulfuricurvum, Sulfurovum, Gallionella, Sideroxydans and Ferriphaselus say?

##**3 -** Genomic information for the most abundant genera in the dataset:

#### **Gallionella:**

Gallionella is an microaerophilic iron-oxidiser that uses $Fe^{+2}$ solely as electron donor. It's equipped for aerotaxis and motile. Grows preferrably at lower temperatures (6¬∫C) and is unable to grow at 30¬∫C. Carbon metabolism works through the RubisCO pathway.

```{r echo=FALSE}
gal_path = "/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Gallionella_Sideroxydans_genomes/Gallionella_genome_Emerson_2013.png"
include_graphics(gal_path)
```

#### **Sideroxydans:**

Sideroxydans is a motile microaerophilic iron-oxidiser that uses $Fe^{+2}$ as electron donor but also has the potential to use thiosulfate. This is due to the presence of a *sox* operon (ABXYZ) in its genome. Grows preferrably at higher temperatures (30¬∫C) and is unable to grow at 6¬∫C. Carbon metabolism works through the RubisCO pathway.

```{r echo=FALSE}
sid_path ="/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Gallionella_Sideroxydans_genomes/Sideroxydans_genome_Emerson_2013.png"
include_graphics(sid_path)
```

#### **Ferriphaselus:**

Ferriphaselus is a motile microaerophilic iron-oxidiser that uses $Fe^{+2}$ solely as electron donor. However, it lacks the *Mto* gene cluster to oxidise $Fe^{+2}$ and seems to use instead a panoply of alternative FeO metabolisms specially including *act* (AB1B2CDEF). Further studies are needed to prove its capacity to grow on sulfur species, since not only it has a *sox* operon (ABXYZ) in its genome but also *dsr* (ABEFHCMKJOPN), *rdsr* (DAGC), *Sqr* and *soe* (ABC). Grows preferrably at higher temperatures (30¬∫C) and is unable to grow at 6¬∫C. Carbon metabolism works through the RubisCO pathway.

#### **Sulfuricurvum:**

Sulfuricurvum is a motile facultative anaerobe that can use sulfide, sulfite and elemental sulfur as electron donors along with oxygen, nitrate and nitric oxide as electron acceptors. To allow that number of electron donors, Sulfuricurvum not only uses the *sox* (ABXYZ) operon, but also *Sqr* (DF), *Sor* (AB) and *Fcc* (AB). 4 plasmids were found in Handley *et al.* (2014) with one associated to nitrogen metabolism. Carbon metabolism works through the rTCA cycle.

```{r echo=FALSE}
sid_path ="/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Sulfurovum_Sulfuricurvum_genomes/Sulfuricurvum_genome_Handley_2014.png"
include_graphics(sid_path)
```

#### **Sulfurovum:**

Sulfurovum is a non-motile facultative anaerobe that can use thiosulfate and elemental sulfur as electron donors along with oxygen and nitrate  as electron acceptors. Sulfurovum only uses the *sox* (ABXYZ) operon, as opposed to Sulfuricurvum. Carbon metabolism works through the rTCA cycle.

### Genomic hits for FeOB metal resistance potential:

**Genome Accessions:** NC_014394.1, NC_013959.1, BBTH01000001:BBTH01000023, NZ_CP011308.1, CP003920.1.

Genome annotation done with ```PROKKA v1.12``` (https://github.com/tseemann/prokka), functional annotation with ```EggNOG-mapper``` (http://eggnogdb.embl.de/app/emapper#/app/emapper).

```{r echo=FALSE}
Gal = c(3,3,14,3,1,1,1,0,15,3,6)
Sid = c(0,2,4,2,5,1,1,1,5,2,6)
Fer = c(0,4,6,2,2,0,2,2,7,2,4)
Sulfuric = c(0,3,4,2,3,1,0,0,2,2,1)
Sulfurov = c(0,3,5,4,6,1,2,1,3,3,6)
Els = c("Mercury","Arsenic","Cobalt","Zinc","Copper","Manganese","Selenium","Magnesium","Heavy metal efflux pump/-related
","Mg/Co Transport","Sodium exchange-related")

g_hits_df = data.frame("Hits" = Els, 
                       "Gallionella" = Gal, 
                       "Sideroxydans" = Sid, 
                       "Ferriphaselus" = Fer,
                       "Sulfuricurvum" = Sulfuric,
                       "Sulfurovum" = Sulfurov)
options(knitr.table.format = "html") 

g_hits_df %>%
mutate_if(is.numeric, function(x) {
cell_spec(x, "html", bold = T, color = spec_color(x, end = 0.9),
          font_size = spec_font_size(x))
  }) %>%
kable("html", escape = F, align = "c") %>%
  kable_styling(bootstrap_options = c("striped"),
                full_width = F)
```

As advertised, *Gallionella* seems to be a lot more prepared to thrive in heavy-metal-rich environments. Maybe that gives this genus an advantage in colonizing most of the iron-rich sites in South Wales. Not a lot can be taken from this since these genomes were generated from far away sites across the world. They do, however, reveal the potential in some of these genera for competitive advantages in heavy-metal-rich sites.

Ok then, so do any of the metals we measured have any impact on the relative abundance of any of these genera?

##**4 -** Metals and relative abundance of important genera:

Can we try and get numbers out of these supposed relations?

What follows is a heatmap of *Pearson* correlations. *Pearson* assumes linearity between variables as well as normal distributions of data and is sensible to transformation. Normalization of the metadata variables to a 0-1 scale didn't change results here.
```{r echo=FALSE, message=FALSE, warning=FALSE}
metadata_cor_main_genera = ps.b.deep_128.glom.df[, c("Genus", "Abundance", "Site.name", "Month", "Time_point","SO4..¬µg.ml",
                                                     "DO..sat","Temp.C","pH",
                                                     "X55.Mn.ng.ml","Mg.¬µg.ml","X68.Zn.ng.ml","Na.¬µg.ml","X6.Li.ng.ml",
                                                     "X135.Ba.ng.ml","X11.B.ng.ml","X75.As..He..ng.ml","X59.Co.ng.ml",
                                                     "X63.Cu.ng.ml","X78.Se.ng.ml","X200.Hg.ng.ml","X201.Hg.ng.ml","X202.Hg.ng.ml")]
names(metadata_cor_main_genera) = c("Genus", "Abundance", "Site_name", "Month", 
                                    "Time_point","SO4","DO","Temp","pH"
                                    ,"Mn","Mg","Zn","Na","Li","Ba",
                                    "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")

cor_genera_geochem = as(metadata_cor_main_genera[, c("Genus", "Abundance", "Site_name", "Month",
                                                     "Temp","SO4","DO","pH",
                                                     "Mn","Mg","Zn","Na","Li","Ba",
                                                     "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")], "data.frame")
cor_genera_geochem_gal = subset(cor_genera_geochem, Genus=="Gallionella")
cor_genera_geochem_sid = subset(cor_genera_geochem, Genus=="Sideroxydans")
cor_genera_geochem_fer = subset(cor_genera_geochem, Genus=="Ferriphaselus")
cor_genera_geochem_sulfuri = subset(cor_genera_geochem, Genus=="Sulfuricurvum")
cor_genera_geochem_sulfuro = subset(cor_genera_geochem, Genus=="Sulfurovum")

cor_genera_geochem_gal <- cor_genera_geochem_gal[, -which(names(cor_genera_geochem_gal) %in% c("Genus","Site_name","Month"))]
cor_genera_geochem_gal_ggcorr<-ggcorr(cor_genera_geochem_gal, 
                                           method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, 
                                          label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn",
                                           nbreaks = 10, hjust = 1, 
                                           size=4, layout.exp = 2)

cor_genera_geochem_sid <- cor_genera_geochem_sid[, -which(names(cor_genera_geochem_sid) %in% c("Genus","Site_name","Month"))]
cor_genera_geochem_sid_ggcorr<-ggcorr(cor_genera_geochem_sid, 
                                           method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, 
                                          label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn",
                                           nbreaks = 10, hjust = 1, 
                                           size=4, layout.exp = 2)

cor_genera_geochem_fer <- cor_genera_geochem_fer[, -which(names(cor_genera_geochem_fer) %in% c("Genus","Site_name","Month"))]
cor_genera_geochem_fer_ggcorr<-ggcorr(cor_genera_geochem_fer, 
                                           method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, 
                                           label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn",
                                           nbreaks = 10, hjust = 1, 
                                           size=4, layout.exp = 2)

cor_genera_geochem_sulfuri <- cor_genera_geochem_sulfuri[, -which(names(cor_genera_geochem_sulfuri) %in% c("Genus","Site_name","Month"))]
cor_genera_geochem_sulfuri_ggcorr<-ggcorr(cor_genera_geochem_sulfuri, 
                                           method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, 
                                          label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn",
                                           nbreaks = 10, hjust = 1, 
                                           size=4, layout.exp = 2)

cor_genera_geochem_sulfuro <- cor_genera_geochem_sulfuro[, -which(names(cor_genera_geochem_sulfuro) %in% c("Genus","Site_name","Month"))]
cor_genera_geochem_sulfuro_ggcorr<-ggcorr(cor_genera_geochem_sulfuro, 
                                           method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, 
                                          label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn",
                                           nbreaks = 10, hjust = 1, 
                                           size=4, layout.exp = 2)

#THIS IS SO LAZY

pearson_gal = head(cor_genera_geochem_gal_ggcorr$data, n=18)
pearson_gal$y = "Gallionella"
pearson_gal$x = c("Temp","SO4","DO","pH",
                  "Mn","Mg","Zn","Na","Li","Ba",
                  "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
pearson_sid = head(cor_genera_geochem_sid_ggcorr$data, n=18)
pearson_sid$y = "Sideroxydans"
pearson_sid$x = c("Temp","SO4","DO","pH",
                  "Mn","Mg","Zn","Na","Li","Ba",
                  "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
pearson_fer = head(cor_genera_geochem_fer_ggcorr$data, n=18)
pearson_fer$y = "Ferriphaselus"
pearson_fer$x = c("Temp","SO4","DO","pH",
                  "Mn","Mg","Zn","Na","Li","Ba",
                  "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
pearson_sulfuri = head(cor_genera_geochem_sulfuri_ggcorr$data, n=18)
pearson_sulfuri$y = "Sulfuricurvum"
pearson_sulfuri$x = c("Temp","SO4","DO","pH",
                  "Mn","Mg","Zn","Na","Li","Ba",
                  "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
pearson_sulfuro = head(cor_genera_geochem_sulfuro_ggcorr$data, n=18)
pearson_sulfuro$y = "Sulfurovum"
pearson_sulfuro$x = c("Temp","SO4","DO","pH",
                  "Mn","Mg","Zn","Na","Li","Ba",
                  "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")

#this is the correlation coefs table done. now for significance
#help from https://stackoverflow.com/questions/12196756/significance-level-added-to-matrix-correlation-heatmap-using-ggplot2

# Function to get the probability into a whole matrix not half, here is Spearman you can change it to Kendall or Pearson
cor.prob.all <- function (X, dfr = nrow(X) - 2) {
R <- cor(X, use="complete.obs",method="spearman")
r2 <- R^2
Fstat <- r2 * dfr/(1 - r2)
R<- 1 - pf(Fstat, 1, dfr)
R[row(R) == col(R)] <- NA
R
}

cor_genera_geochem_gal_p=cor.prob.all(cor_genera_geochem_gal)
cor_genera_geochem_sid_p=cor.prob.all(cor_genera_geochem_sid)
cor_genera_geochem_fer_p=cor.prob.all(cor_genera_geochem_fer)
cor_genera_geochem_sulfuri_p=cor.prob.all(cor_genera_geochem_sulfuri)
cor_genera_geochem_sulfuro_p=cor.prob.all(cor_genera_geochem_sulfuro)

cor_genera_geochem_gal_p <- data.frame(row=rownames(cor_genera_geochem_gal_p),cor_genera_geochem_gal_p) # create a column called "row" 
rownames(cor_genera_geochem_gal_p) <- NULL
cor_genera_geochem_gal_p.m <- head(melt(cor_genera_geochem_gal_p),n=19)[-1,]
cor_genera_geochem_gal_p.m$value2<-cut(cor_genera_geochem_gal_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_genera_geochem_gal_p.m[is.na(cor_genera_geochem_gal_p.m)] <- ""
gal_pearson_pvalue<-cbind(pearson_gal,cor_genera_geochem_gal_p.m)
cor_genera_geochem_gal <- gal_pearson_pvalue[, -which(names(gal_pearson_pvalue) %in% c("row","variable"))]
names(cor_genera_geochem_gal)[6]<-paste("valuep")
names(cor_genera_geochem_gal)[7]<-paste("signif.")

cor_genera_geochem_sid_p <- data.frame(row=rownames(cor_genera_geochem_sid_p),cor_genera_geochem_sid_p) # create a column called "row" 
rownames(cor_genera_geochem_sid_p) <- NULL
cor_genera_geochem_sid_p.m <- head(melt(cor_genera_geochem_sid_p),n=19)[-1,]
cor_genera_geochem_sid_p.m$value2<-cut(cor_genera_geochem_sid_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_genera_geochem_sid_p.m[is.na(cor_genera_geochem_sid_p.m)] <- ""
sid_pearson_pvalue<-cbind(pearson_sid,cor_genera_geochem_sid_p.m)
cor_genera_geochem_sid <- sid_pearson_pvalue[, -which(names(sid_pearson_pvalue) %in% c("row","variable"))]
names(cor_genera_geochem_sid)[6]<-paste("valuep")
names(cor_genera_geochem_sid)[7]<-paste("signif.")

cor_genera_geochem_fer_p <- data.frame(row=rownames(cor_genera_geochem_fer_p),cor_genera_geochem_fer_p) # create a column called "row" 
rownames(cor_genera_geochem_fer_p) <- NULL
cor_genera_geochem_fer_p.m <- head(melt(cor_genera_geochem_fer_p),n=19)[-1,]
cor_genera_geochem_fer_p.m$value2<-cut(cor_genera_geochem_fer_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_genera_geochem_fer_p.m[is.na(cor_genera_geochem_fer_p.m)] <- ""
fer_pearson_pvalue<-cbind(pearson_fer,cor_genera_geochem_fer_p.m)
cor_genera_geochem_fer <- fer_pearson_pvalue[, -which(names(fer_pearson_pvalue) %in% c("row","variable"))]
names(cor_genera_geochem_fer)[6]<-paste("valuep")
names(cor_genera_geochem_fer)[7]<-paste("signif.")

cor_genera_geochem_sulfuri_p <- data.frame(row=rownames(cor_genera_geochem_sulfuri_p),cor_genera_geochem_sulfuri_p) # create a column called "row" 
rownames(cor_genera_geochem_sulfuri_p) <- NULL
cor_genera_geochem_sulfuri_p.m <- head(melt(cor_genera_geochem_sulfuri_p),n=19)[-1,]
cor_genera_geochem_sulfuri_p.m$value2<-cut(cor_genera_geochem_sulfuri_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_genera_geochem_sulfuri_p.m[is.na(cor_genera_geochem_sulfuri_p.m)] <- ""
sulfuri_pearson_pvalue<-cbind(pearson_sulfuri,cor_genera_geochem_sulfuri_p.m)
cor_genera_geochem_sulfuri <- sulfuri_pearson_pvalue[, -which(names(sulfuri_pearson_pvalue) %in% c("row","variable"))]
names(cor_genera_geochem_sulfuri)[6]<-paste("valuep")
names(cor_genera_geochem_sulfuri)[7]<-paste("signif.")

cor_genera_geochem_sulfuro_p <- data.frame(row=rownames(cor_genera_geochem_sulfuro_p),cor_genera_geochem_sulfuro_p) # create a column called "row" 
rownames(cor_genera_geochem_sulfuro_p) <- NULL
cor_genera_geochem_sulfuro_p.m <- head(melt(cor_genera_geochem_sulfuro_p),n=19)[-1,]
cor_genera_geochem_sulfuro_p.m$value2<-cut(cor_genera_geochem_sulfuro_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_genera_geochem_sulfuro_p.m[is.na(cor_genera_geochem_sulfuro_p.m)] <- ""
sulfuro_pearson_pvalue<-cbind(pearson_sulfuro,cor_genera_geochem_sulfuro_p.m)
cor_genera_geochem_sulfuro <- sulfuro_pearson_pvalue[, -which(names(sulfuro_pearson_pvalue) %in% c("row","variable"))]
names(cor_genera_geochem_sulfuro)[6]<-paste("valuep")
names(cor_genera_geochem_sulfuro)[7]<-paste("signif.")

#
replot_pearson = rbind(cor_genera_geochem_gal,cor_genera_geochem_sid,cor_genera_geochem_fer,
                       cor_genera_geochem_sulfuri,cor_genera_geochem_sulfuro)
# replot_pearson$x<-pearson_gal$x


low="#af8dc3"
mid="#f7f7f7"
high="#7fbf7b"

replot_pearson$x = factor(replot_pearson$x,levels = c("Temp","DO","pH","SO4","As","B","Ba","Co","Cu",
                                                      "Li","Mg","Mn","Na","Se","Zn","200_Hg","201_Hg","202_Hg"))
replot_pearson$y = factor(replot_pearson$y,levels = c("Sulfurovum","Sulfuricurvum",
                                                      "Ferriphaselus","Sideroxydans","Gallionella"))
#maybe keeping tile size proportional to pvalue?
# replot_pearson$trans_valuep <- abs(log(replot_pearson$valuep))/45
# replot_pearson$trans_valuep[which(!is.finite(replot_pearson$trans_valuep))] <- 0
  # geom_tile(aes(fill=breaks, height=trans_valuep, width=trans_valuep)) +
  # scale_size_manual(values=c(1,3,5,7),
  #                    labels=c("<0.001", "<0.01", "<0.05",NA)) +

replot_pearson_plot <- ggplot(replot_pearson,
       aes(x, y)) +
  geom_tile(aes(fill=breaks, width=0.8, height=0.8)) +
  geom_tile(data=replot_pearson[!is.na(replot_pearson$signif.),],
            aes(color=signif., width=0.9, height=0.9), fill="transparent", size=1) +
  coord_equal() +
  geom_text(aes(label = label,
                alpha = abs(coefficient)), 
            size = 4, color = "black") +
  scale_fill_manual(values = colorRampPalette(c(low, mid, high))(length(levels(replot_pearson$breaks))),
                    drop = FALSE, 
                    labels = c("-1, -0.8","-0.8, -0.6","-0.6, -0.4","-0.4, -0.2","-0.2, 0",
                               "0, 0.2","0.2, 0.4","0.4, 0.6","0.6, 0.8","0.8, 1")) +
  scale_color_manual(values=c("gray30","gray63","gray87"),
                     labels=c("<0.001", "<0.01", "<0.05")) +
  labs(title = "Pearson coefficients",
       subtitle = "Using non-transformed relative abundances and metadata variables" ) +
  theme(title = element_text(hjust = 0),
        axis.title.x = element_blank(),
        axis.text.x = element_text(angle=30, hjust = 1),
        axis.title.y = element_blank(),
        axis.text.y = element_text(face = "italic")) +
  guides(alpha = FALSE,
         fill = guide_legend(title = "r",
                             reverse = TRUE,
                             title.theme = element_text(face="italic",
                                                        angle = 0)),
         color= guide_legend(title="Significance",
                             override.aes = list(fill="white")))
```
```{r echo=FALSE, fig.align="center", fig.width=12, fig.height=4}
replot_pearson_plot
```

So from *Pearson* correlations it looks like each of the relative abundance of each FeOB is defined by a specific set of metadata variables that probably drives co-occurrence across the dataset. 

**Gallionella** is very positively correlated to cobalt concentrations and negatively to barium, temperature, pH and sodium concentrations. The first two make sense since this genus is known to grow better a lower temperatures and has had its potential for manganese, cobalt and arsenic resistance reported [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3770913/).

Regarding **Sideroxydans**, it seems the only meaningful correlations point positively to cobalt and copper negatively to pH and $Hg^{200}$. Again, at least some part of this may have been predicted by [previous work](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3770913/). It's positively correlated to $SO^{4}$, which might indicate the presence of a *sox* operon as in the genome analysed previously.

**Ferriphaselus** stands out as being positively correlated to cobalt, like **Gallionella**, and negatively correlated with Na and Mg.

A number of correlations arise with **Sulfuricurvum**, where it seems that it's strongly associated to dissolved oxygen, boron, lithium and mercury concentrations. It's negatively associated to the presence of copper and selenium.

**Sulfurovum** presence is by positive correlations towards temperature, pH, B, Mg, and $Hg^{200,201}$. It's also strongly negatively correlated to copper and manganese presence.

##**5 -** Where do CPR taxa fit in the dataset?

```{r cpr plot, echo=FALSE, message=FALSE, warning=FALSE}
ps.b.cpr = subset_taxa(ps.b.r, 
                            Phylum=="Omnitrophica" | Phylum=="Parcubacteria")
ps.b.cpr.glom <- tax_glom(ps.b.cpr, taxrank = 'Class', NArm = FALSE)
ps.b.cpr.glom.df <- data.table(psmelt(ps.b.cpr.glom))
ps.b.cpr.glom.df$Class <- as.character(ps.b.cpr.glom.df$Class)
ps.b.cpr.glom.df$Phylum <- as.character(ps.b.cpr.glom.df$Phylum)
ps.b.cpr.glom.df[, median := median(Abundance, na.rm = TRUE), 
                        by = "Class"]
ps.b.cpr.glom.df[(median <= 0.001 & Phylum == "Parcubacteria"), Class := "Other Parcubacteria"]
ps.b.cpr.glom.df[(median <= 0.001 & Phylum == "Omnitrophica"), Class := "Other Omnitrophica"]
ps.b.cpr.glom.df$Class[is.na(ps.b.cpr.glom.df$Class) & ps.b.cpr.glom.df$Phylum=="Omnitrophica"] <- "Unc. Omnitrophica"
ps.b.cpr.glom.df$Class[is.na(ps.b.cpr.glom.df$Class) & ps.b.cpr.glom.df$Phylum=="Parcubacteria"] <- "Unc. Parcubacteria"
ps.b.cpr.glom.df <- data.frame(lapply(ps.b.cpr.glom.df, 
                                               function(x) {gsub("Candidatus_", "Cand. ", x)}))
ps.b.cpr.glom.df <- data.frame(lapply(ps.b.cpr.glom.df, 
                                               function(x) {gsub("Omnitrophica_Incertae_Sedis", "Omnitrophica Inc. Sed.", x)}))
ps.b.cpr.glom.df$Site.name = factor(ps.b.cpr.glom.df$Site.name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
metadata_cor_cpr = ps.b.cpr.glom.df[, c("Class", "Abundance", "Site.name", "Month", "Time_point","SO4..¬µg.ml",
                                                     "DO..sat","Temp.C","pH",
                                                     "X55.Mn.ng.ml","Mg.¬µg.ml","X68.Zn.ng.ml","Na.¬µg.ml","X6.Li.ng.ml",
                                                     "X135.Ba.ng.ml","X11.B.ng.ml","X75.As..He..ng.ml","X59.Co.ng.ml",
                                                     "X63.Cu.ng.ml","X78.Se.ng.ml","X200.Hg.ng.ml","X201.Hg.ng.ml","X202.Hg.ng.ml")]
names(metadata_cor_cpr) = c("Class", "Abundance", "Site_name", "Month", 
                                    "Time_point","SO4","DO","Temp","pH"
                                    ,"Mn","Mg","Zn","Na","Li","Ba",
                                    "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
cor_cpr_geochem = as(metadata_cor_cpr[, c("Class", "Abundance", "Site_name", "Month",
                                                     "Temp","SO4","DO","pH",
                                                     "Mn","Mg","Zn","Na","Li","Ba",
                                                     "B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")], "data.frame")

cor_cpr_geochem_unc_omni = subset(cor_cpr_geochem, Class=="Unc. Omnitrophica")
cor_cpr_geochem_unc_parc = subset(cor_cpr_geochem, Class=="Unc. Parcubacteria")
cor_cpr_geochem_azamb = subset(cor_cpr_geochem, Class=="Cand. Azambacteria")
cor_cpr_geochem_jorge = subset(cor_cpr_geochem, Class=="Cand. Jorgensenbacteria")
cor_cpr_geochem_nomur = subset(cor_cpr_geochem, Class=="Cand. Nomurabacteria")
cor_cpr_geochem_moran = subset(cor_cpr_geochem, Class=="Cand. Moranbacteria")
cor_cpr_geochem_omni_is = subset(cor_cpr_geochem, Class=="Omnitrophica Inc. Sed.")
cor_cpr_geochem_falko = subset(cor_cpr_geochem, Class=="Cand. Falkowbacteria")
cor_cpr_geochem_wolfe = subset(cor_cpr_geochem, Class=="Cand. Wolfebacteria")
cor_cpr_geochem_magas = subset(cor_cpr_geochem, Class=="Cand. Magasanikbacteria")
cor_cpr_geochem_ot_parc = subset(cor_cpr_geochem, Class=="Other Parcubacteria")

cor_cpr_geochem_unc_omni <- cor_cpr_geochem_unc_omni[, -which(names(cor_cpr_geochem_unc_omni) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_unc_omni <- sapply( cor_cpr_geochem_unc_omni, as.numeric )
cor_cpr_geochem_unc_omni_ggcorr<-ggcorr(cor_cpr_geochem_unc_omni, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_unc_parc <- cor_cpr_geochem_unc_parc[, -which(names(cor_cpr_geochem_unc_parc) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_unc_parc <- sapply( cor_cpr_geochem_unc_parc, as.numeric )
cor_cpr_geochem_unc_parc_ggcorr<-ggcorr(cor_cpr_geochem_unc_parc, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_azamb <- cor_cpr_geochem_azamb[, -which(names(cor_cpr_geochem_azamb) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_azamb <- sapply( cor_cpr_geochem_azamb, as.numeric )
cor_cpr_geochem_azamb_ggcorr<-ggcorr(cor_cpr_geochem_azamb, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_jorge <- cor_cpr_geochem_jorge[, -which(names(cor_cpr_geochem_jorge) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_jorge <- sapply( cor_cpr_geochem_jorge, as.numeric )
cor_cpr_geochem_jorge_ggcorr<-ggcorr(cor_cpr_geochem_jorge, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_nomur <- cor_cpr_geochem_nomur[, -which(names(cor_cpr_geochem_nomur) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_nomur <- sapply( cor_cpr_geochem_nomur, as.numeric )
cor_cpr_geochem_nomur_ggcorr<-ggcorr(cor_cpr_geochem_nomur, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_moran <- cor_cpr_geochem_moran[, -which(names(cor_cpr_geochem_moran) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_moran <- sapply( cor_cpr_geochem_moran, as.numeric )
cor_cpr_geochem_moran_ggcorr<-ggcorr(cor_cpr_geochem_moran, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_omni_is <- cor_cpr_geochem_omni_is[, -which(names(cor_cpr_geochem_omni_is) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_omni_is <- sapply( cor_cpr_geochem_omni_is, as.numeric )
cor_cpr_geochem_omni_is_ggcorr<-ggcorr(cor_cpr_geochem_omni_is, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_falko <- cor_cpr_geochem_falko[, -which(names(cor_cpr_geochem_falko) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_falko <- sapply( cor_cpr_geochem_falko, as.numeric )
cor_cpr_geochem_falko_ggcorr<-ggcorr(cor_cpr_geochem_falko, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_wolfe <- cor_cpr_geochem_wolfe[, -which(names(cor_cpr_geochem_wolfe) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_wolfe <- sapply( cor_cpr_geochem_wolfe, as.numeric )
cor_cpr_geochem_wolfe_ggcorr<-ggcorr(cor_cpr_geochem_wolfe, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_magas <- cor_cpr_geochem_magas[, -which(names(cor_cpr_geochem_magas) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_magas <- sapply( cor_cpr_geochem_magas, as.numeric )
cor_cpr_geochem_magas_ggcorr<-ggcorr(cor_cpr_geochem_magas, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)

cor_cpr_geochem_ot_parc <- cor_cpr_geochem_ot_parc[, -which(names(cor_cpr_geochem_ot_parc) %in% c("Class","Site_name","Month"))]
cor_cpr_geochem_ot_parc <- sapply( cor_cpr_geochem_ot_parc, as.numeric )
cor_cpr_geochem_ot_parc_ggcorr<-ggcorr(cor_cpr_geochem_ot_parc, method = c("complete.obs", "spearman"),
                                           label = TRUE, label_size = 3, label_round = 2, label_alpha = TRUE,
                                           palette = "PRGn", nbreaks = 10, hjust = 1, size=4, layout.exp = 2)
labels=c("Temp","SO4","DO","pH","Mn","Mg","Zn","Na","Li","Ba","B","As","Co","Cu","Se","200_Hg","201_Hg","202_Hg")
pearson_unc_omni = head(cor_cpr_geochem_unc_omni_ggcorr$data, n=18)
pearson_unc_omni$y = "Unc. Omnitrophica"
pearson_unc_omni$x = labels
pearson_unc_parc = head(cor_cpr_geochem_unc_parc_ggcorr$data, n=18)
pearson_unc_parc$y = "Unc. Parcubacteria"
pearson_unc_parc$x = labels
pearson_azamb = head(cor_cpr_geochem_azamb_ggcorr$data, n=18)
pearson_azamb$y = "Cand. Azambacteria"
pearson_azamb$x = labels
pearson_jorge = head(cor_cpr_geochem_jorge_ggcorr$data, n=18)
pearson_jorge$y = "Cand. Jorgensenbacteria"
pearson_jorge$x = labels
pearson_nomur = head(cor_cpr_geochem_nomur_ggcorr$data, n=18)
pearson_nomur$y = "Cand. Nomurabacteria"
pearson_nomur$x = labels
pearson_moran = head(cor_cpr_geochem_moran_ggcorr$data, n=18)
pearson_moran$y = "Cand. Moranbacteria"
pearson_moran$x = labels
pearson_omni_is = head(cor_cpr_geochem_omni_is_ggcorr$data, n=18)
pearson_omni_is$y = "Omnitrophica Inc. Sed."
pearson_omni_is$x = labels
pearson_falko = head(cor_cpr_geochem_falko_ggcorr$data, n=18)
pearson_falko$y = "Cand. Falkowbacteria"
pearson_falko$x = labels
pearson_wolfe = head(cor_cpr_geochem_wolfe_ggcorr$data, n=18)
pearson_wolfe$y = "Cand. Wolfebacteria"
pearson_wolfe$x = labels
pearson_magas = head(cor_cpr_geochem_magas_ggcorr$data, n=18)
pearson_magas$y = "Cand. Magasanikbacteria"
pearson_magas$x = labels
pearson_ot_parc = head(cor_cpr_geochem_ot_parc_ggcorr$data, n=18)
pearson_ot_parc$y = "Other Parcubacteria"
pearson_ot_parc$x = labels

cor_cpr_geochem_unc_omni_p=cor.prob.all(cor_cpr_geochem_unc_omni)
cor_cpr_geochem_unc_parc_p=cor.prob.all(cor_cpr_geochem_unc_parc)
cor_cpr_geochem_azamb_p=cor.prob.all(cor_cpr_geochem_azamb)
cor_cpr_geochem_jorge_p=cor.prob.all(cor_cpr_geochem_jorge)
cor_cpr_geochem_nomur_p=cor.prob.all(cor_cpr_geochem_nomur)
cor_cpr_geochem_moran_p=cor.prob.all(cor_cpr_geochem_moran)
cor_cpr_geochem_omni_is_p=cor.prob.all(cor_cpr_geochem_omni_is)
cor_cpr_geochem_falko_p=cor.prob.all(cor_cpr_geochem_falko)
cor_cpr_geochem_wolfe_p=cor.prob.all(cor_cpr_geochem_wolfe)
cor_cpr_geochem_magas_p=cor.prob.all(cor_cpr_geochem_magas)
cor_cpr_geochem_ot_parc_p=cor.prob.all(cor_cpr_geochem_ot_parc)

cor_cpr_geochem_unc_omni_p <- data.frame(row=rownames(cor_cpr_geochem_unc_omni_p),cor_cpr_geochem_unc_omni_p) # create a column called "row" 
rownames(cor_cpr_geochem_unc_omni_p) <- NULL
cor_cpr_geochem_unc_omni_p.m <- head(melt(cor_cpr_geochem_unc_omni_p),n=19)[-1,]
cor_cpr_geochem_unc_omni_p.m$value2<-cut(cor_cpr_geochem_unc_omni_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_unc_omni_p.m[is.na(cor_cpr_geochem_unc_omni_p.m)] <- ""
unc_omni_pearson_pvalue<-cbind(pearson_unc_omni,cor_cpr_geochem_unc_omni_p.m)
cor_cpr_geochem_unc_omni <- unc_omni_pearson_pvalue[, -which(names(unc_omni_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_unc_omni)[6]<-paste("valuep")
names(cor_cpr_geochem_unc_omni)[7]<-paste("signif.")

cor_cpr_geochem_unc_parc_p <- data.frame(row=rownames(cor_cpr_geochem_unc_parc_p),cor_cpr_geochem_unc_parc_p) # create a column called "row" 
rownames(cor_cpr_geochem_unc_parc_p) <- NULL
cor_cpr_geochem_unc_parc_p.m <- head(melt(cor_cpr_geochem_unc_parc_p),n=19)[-1,]
cor_cpr_geochem_unc_parc_p.m$value2<-cut(cor_cpr_geochem_unc_parc_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_unc_parc_p.m[is.na(cor_cpr_geochem_unc_parc_p.m)] <- ""
unc_parc_pearson_pvalue<-cbind(pearson_unc_parc,cor_cpr_geochem_unc_parc_p.m)
cor_cpr_geochem_unc_parc <- unc_parc_pearson_pvalue[, -which(names(unc_parc_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_unc_parc)[6]<-paste("valuep")
names(cor_cpr_geochem_unc_parc)[7]<-paste("signif.")

cor_cpr_geochem_azamb_p <- data.frame(row=rownames(cor_cpr_geochem_azamb_p),cor_cpr_geochem_azamb_p) # create a column called "row" 
rownames(cor_cpr_geochem_azamb_p) <- NULL
cor_cpr_geochem_azamb_p.m <- head(melt(cor_cpr_geochem_azamb_p),n=19)[-1,]
cor_cpr_geochem_azamb_p.m$value2<-cut(cor_cpr_geochem_azamb_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_azamb_p.m[is.na(cor_cpr_geochem_azamb_p.m)] <- ""
azamb_pearson_pvalue<-cbind(pearson_azamb,cor_cpr_geochem_azamb_p.m)
cor_cpr_geochem_azamb <- azamb_pearson_pvalue[, -which(names(azamb_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_azamb)[6]<-paste("valuep")
names(cor_cpr_geochem_azamb)[7]<-paste("signif.")

cor_cpr_geochem_jorge_p <- data.frame(row=rownames(cor_cpr_geochem_jorge_p),cor_cpr_geochem_jorge_p) # create a column called "row" 
rownames(cor_cpr_geochem_jorge_p) <- NULL
cor_cpr_geochem_jorge_p.m <- head(melt(cor_cpr_geochem_jorge_p),n=19)[-1,]
cor_cpr_geochem_jorge_p.m$value2<-cut(cor_cpr_geochem_jorge_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_jorge_p.m[is.na(cor_cpr_geochem_jorge_p.m)] <- ""
jorge_pearson_pvalue<-cbind(pearson_jorge,cor_cpr_geochem_jorge_p.m)
cor_cpr_geochem_jorge <- jorge_pearson_pvalue[, -which(names(jorge_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_jorge)[6]<-paste("valuep")
names(cor_cpr_geochem_jorge)[7]<-paste("signif.")

cor_cpr_geochem_nomur_p <- data.frame(row=rownames(cor_cpr_geochem_nomur_p),cor_cpr_geochem_nomur_p) # create a column called "row" 
rownames(cor_cpr_geochem_nomur_p) <- NULL
cor_cpr_geochem_nomur_p.m <- head(melt(cor_cpr_geochem_nomur_p),n=19)[-1,]
cor_cpr_geochem_nomur_p.m$value2<-cut(cor_cpr_geochem_nomur_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_nomur_p.m[is.na(cor_cpr_geochem_nomur_p.m)] <- ""
nomur_pearson_pvalue<-cbind(pearson_nomur,cor_cpr_geochem_nomur_p.m)
cor_cpr_geochem_nomur <- nomur_pearson_pvalue[, -which(names(nomur_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_nomur)[6]<-paste("valuep")
names(cor_cpr_geochem_nomur)[7]<-paste("signif.")

cor_cpr_geochem_moran_p <- data.frame(row=rownames(cor_cpr_geochem_moran_p),cor_cpr_geochem_moran_p) # create a column called "row" 
rownames(cor_cpr_geochem_moran_p) <- NULL
cor_cpr_geochem_moran_p.m <- head(melt(cor_cpr_geochem_moran_p),n=19)[-1,]
cor_cpr_geochem_moran_p.m$value2<-cut(cor_cpr_geochem_moran_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_moran_p.m[is.na(cor_cpr_geochem_moran_p.m)] <- ""
moran_pearson_pvalue<-cbind(pearson_moran,cor_cpr_geochem_moran_p.m)
cor_cpr_geochem_moran <- moran_pearson_pvalue[, -which(names(moran_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_moran)[6]<-paste("valuep")
names(cor_cpr_geochem_moran)[7]<-paste("signif.")

cor_cpr_geochem_omni_is_p <- data.frame(row=rownames(cor_cpr_geochem_omni_is_p),cor_cpr_geochem_omni_is_p) # create a column called "row" 
rownames(cor_cpr_geochem_omni_is_p) <- NULL
cor_cpr_geochem_omni_is_p.m <- head(melt(cor_cpr_geochem_omni_is_p),n=19)[-1,]
cor_cpr_geochem_omni_is_p.m$value2<-cut(cor_cpr_geochem_omni_is_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_omni_is_p.m[is.na(cor_cpr_geochem_omni_is_p.m)] <- ""
omni_is_pearson_pvalue<-cbind(pearson_omni_is,cor_cpr_geochem_omni_is_p.m)
cor_cpr_geochem_omni_is <- omni_is_pearson_pvalue[, -which(names(omni_is_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_omni_is)[6]<-paste("valuep")
names(cor_cpr_geochem_omni_is)[7]<-paste("signif.")

cor_cpr_geochem_falko_p <- data.frame(row=rownames(cor_cpr_geochem_falko_p),cor_cpr_geochem_falko_p) # create a column called "row" 
rownames(cor_cpr_geochem_falko_p) <- NULL
cor_cpr_geochem_falko_p.m <- head(melt(cor_cpr_geochem_falko_p),n=19)[-1,]
cor_cpr_geochem_falko_p.m$value2<-cut(cor_cpr_geochem_falko_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_falko_p.m[is.na(cor_cpr_geochem_falko_p.m)] <- ""
falko_pearson_pvalue<-cbind(pearson_falko,cor_cpr_geochem_falko_p.m)
cor_cpr_geochem_falko <- falko_pearson_pvalue[, -which(names(falko_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_falko)[6]<-paste("valuep")
names(cor_cpr_geochem_falko)[7]<-paste("signif.")

cor_cpr_geochem_wolfe_p <- data.frame(row=rownames(cor_cpr_geochem_wolfe_p),cor_cpr_geochem_wolfe_p) # create a column called "row" 
rownames(cor_cpr_geochem_wolfe_p) <- NULL
cor_cpr_geochem_wolfe_p.m <- head(melt(cor_cpr_geochem_wolfe_p),n=19)[-1,]
cor_cpr_geochem_wolfe_p.m$value2<-cut(cor_cpr_geochem_wolfe_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_wolfe_p.m[is.na(cor_cpr_geochem_wolfe_p.m)] <- ""
wolfe_pearson_pvalue<-cbind(pearson_wolfe,cor_cpr_geochem_wolfe_p.m)
cor_cpr_geochem_wolfe <- wolfe_pearson_pvalue[, -which(names(wolfe_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_wolfe)[6]<-paste("valuep")
names(cor_cpr_geochem_wolfe)[7]<-paste("signif.")

cor_cpr_geochem_magas_p <- data.frame(row=rownames(cor_cpr_geochem_magas_p),cor_cpr_geochem_magas_p) # create a column called "row" 
rownames(cor_cpr_geochem_magas_p) <- NULL
cor_cpr_geochem_magas_p.m <- head(melt(cor_cpr_geochem_magas_p),n=19)[-1,]
cor_cpr_geochem_magas_p.m$value2<-cut(cor_cpr_geochem_magas_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_magas_p.m[is.na(cor_cpr_geochem_magas_p.m)] <- ""
magas_pearson_pvalue<-cbind(pearson_magas,cor_cpr_geochem_magas_p.m)
cor_cpr_geochem_magas <- magas_pearson_pvalue[, -which(names(magas_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_magas)[6]<-paste("valuep")
names(cor_cpr_geochem_magas)[7]<-paste("signif.")

cor_cpr_geochem_ot_parc_p <- data.frame(row=rownames(cor_cpr_geochem_ot_parc_p),cor_cpr_geochem_ot_parc_p) # create a column called "row" 
rownames(cor_cpr_geochem_ot_parc_p) <- NULL
cor_cpr_geochem_ot_parc_p.m <- head(melt(cor_cpr_geochem_ot_parc_p),n=19)[-1,]
cor_cpr_geochem_ot_parc_p.m$value2<-cut(cor_cpr_geochem_ot_parc_p.m$value,
                                       breaks=c(-Inf, 0.001, 0.01, 0.05),label=c("***", "** ", "*  ")) 
cor_cpr_geochem_ot_parc_p.m[is.na(cor_cpr_geochem_ot_parc_p.m)] <- ""
ot_parc_pearson_pvalue<-cbind(pearson_ot_parc,cor_cpr_geochem_ot_parc_p.m)
cor_cpr_geochem_ot_parc <- ot_parc_pearson_pvalue[, -which(names(ot_parc_pearson_pvalue) %in% c("row","variable"))]
names(cor_cpr_geochem_ot_parc)[6]<-paste("valuep")
names(cor_cpr_geochem_ot_parc)[7]<-paste("signif.")

#
replot_cpr_pearson = rbind(cor_cpr_geochem_unc_omni,cor_cpr_geochem_unc_parc,cor_cpr_geochem_azamb,
                       cor_cpr_geochem_jorge,cor_cpr_geochem_nomur, cor_cpr_geochem_moran,
                       cor_cpr_geochem_omni_is, cor_cpr_geochem_falko, cor_cpr_geochem_wolfe,
                       cor_cpr_geochem_magas, cor_cpr_geochem_ot_parc)
# replot_cpr_pearson$x<-pearson_unc_omni$x


low="#af8dc3"
mid="#f7f7f7"
high="#7fbf7b"

replot_cpr_pearson$x = factor(replot_cpr_pearson$x,levels = c("Temp","DO","pH","SO4","As","B","Ba","Co","Cu",
                                                      "Li","Mg","Mn","Na","Se","Zn","200_Hg","201_Hg","202_Hg"))
replot_cpr_pearson$y = factor(replot_cpr_pearson$y,levels = c("Unc. Omnitrophica","Omnitrophica Inc. Sed.",
                                                              "Unc. Parcubacteria","Other Parcubacteria","Cand. Wolfebacteria",
                                                              "Cand. Magasanikbacteria","Cand. Moranbacteria","Cand. Nomurabacteria",
                                                              "Cand. Jorgensenbacteria","Cand. Falkowbacteria","Cand. Azambacteria"))
replot_cpr_pearson_plot <- ggplot(replot_cpr_pearson,
       aes(x, y)) +
  geom_tile(aes(fill=breaks, width=0.8, height=0.8)) +
  geom_tile(data=replot_cpr_pearson[!is.na(replot_cpr_pearson$signif.),],
            aes(color=signif., width=0.9, height=0.9), fill="transparent", size=1) +
  coord_equal() +
  geom_text(aes(label = label,
                alpha = abs(coefficient)), 
            size = 5, color = "black") +
  scale_fill_manual(values = colorRampPalette(c(low, mid, high))(length(levels(replot_cpr_pearson$breaks))),
                    drop = FALSE, 
                    labels = c("-1, -0.8","-0.8, -0.6","-0.6, -0.4","-0.4, -0.2","-0.2, 0",
                               "0, 0.2","0.2, 0.4","0.4, 0.6","0.6, 0.8","0.8, 1")) +
  scale_color_manual(values=c("gray30","gray63","gray87"),
                     labels=c("<0.001", "<0.01", "<0.05")) +
  labs(title = "Pearson coefficients",
       subtitle = "Using non-transformed relative abundances and metadata variables" ) +
  theme(title = element_text(hjust = 0),
        axis.title.x = element_blank(),
        axis.text.x = element_text(size=15,angle=30, hjust = 1),
        axis.title.y = element_blank(),
        axis.text.y = element_text(size=13,
                                   face = "italic")) +
  guides(alpha = FALSE,
         fill = guide_legend(title = "r",
                             reverse = TRUE,
                             title.theme = element_text(face="italic",
                                                        angle = 0)),
         color= guide_legend(title="Significance",
                             override.aes = list(fill="white")))
```
```{r echo=FALSE, fig.align="center", fig.width=14, fig.height=8}
replot_cpr_pearson_plot
```

Even though of course for most of it not as many correlations come up, some interesting ones still do. Namely, most of the CPR phyla show some positive correlation towards arsenic. No general distinctions come up regarding phylum-level taxonomic differences. Temperature doesn't seem to be related at all with CPR relative abundance.

Interestingly, **Cand. Falkowbacteria** has the strongest positive correlation with barium, **Cand. Moranbacteria** with zinc, **Cand. Azambacteria** with lithium and **Omnitrophica Inc. Sedis** with selenium. 

Any ¬¥evidence¬¥ for this in the genomes?

### Genomic hits for CPR metal resistance potential, lifestyles:

**Genome Accessions:** SAMN04314763, MFPS01000001.1, SAMN04314878, SAMN04315031, SAMN04314976, SAMN04315174, SAMN04314587, SAMN04315171.

Genome annotation done with ```PROKKA v1.12``` (https://github.com/tseemann/prokka), functional annotation with ```EggNOG-mapper``` (http://eggnogdb.embl.de/app/emapper#/app/emapper).

```{r echo=FALSE}
Omni = c(0,0,7,2,1,2,0,1,2,1,1,0,2,0)
Azam = c(0,0,5,0,0,0,0,1,0,1,0,0,3,0)
Jorge = c(0,0,0,3,0,0,0,0,0,0,2,0,4,0)
Nomura = c(0,0,0,1,1,1,0,1,2,1,1,0,9,0)
Moran = c(0,0,0,2,2,1,0,1,2,1,0,0,8,0)
Magas = c(0,0,0,3,0,1,0,1,1,1,2,0,9,0)
Wolfe =c(0,0,0,3,1,0,0,0,1,0,1,0,10,0)
Els2 = c("Mercury","Arsenic","Cobalt","Zinc","Copper","Manganese","Selenium","Magnesium","Heavy metal efflux pump/-related
","Mg/Co Transport","Sodium exchange-related","Chemotaxis","Pilli","Flagella")

g_hits_df = data.frame("Hits" = Els2, 
                       "Omnitrophica" = Omni, 
                       "Azambact." = Azam, 
                       "Jorgensenbact." = Jorge,
                       "Nomurabact." = Nomura,
                       "Moranbact." = Moran,
                       "Magasanikbac."=Magas,
                       "Wolfebact."=Wolfe)
options(knitr.table.format = "html") 

g_hits_df %>%
mutate_if(is.numeric, function(x) {
cell_spec(x, "html", bold = T, color = spec_color(x, end = 0.9),
          font_size = spec_font_size(x))
  }) %>%
kable("html", escape = F, align = "c") %>%
  kable_styling(bootstrap_options = c("striped"),
                full_width = F)
```

Seemingly, these genomes seem to be those of either symbionts or parasites, given their general apparent inability to face the high values of heavy metal concentrations in the environment. Omnitrophica and Azambacteria seem to be able to control cobalt efflux, but not much more.

Albeit no chemotaxis or flagella hits were found, hinting at the abovementioned lifestyle hipostheses, genes related to pilli were found.

##**6 -** How do CPR taxa abundances relate to our main genera?

```{r cpr genera plot, message=FALSE, warning=FALSE, include=FALSE}
ps.b.gen = subset_taxa(ps.b.r, 
                      Genus=="Gallionella" | Genus=="Sideroxydans" | Genus=="Ferriphaselus" | Genus=="Sulfurovum" | Genus == "Sulfuricurvum")
ps.b.gen.glom <- tax_glom(ps.b.gen, taxrank = 'Genus', NArm = FALSE)
ps.b.gen.glom.df <- data.table(psmelt(ps.b.gen.glom))
ps.b.gen.glom.df$Class <- as.character(ps.b.gen.glom.df$Class)

ps.b.gen.glom.df[, median := median(Abundance, na.rm = TRUE), 
                        by = "Class"]

ps.b.cpr = subset_taxa(ps.b.r, 
                      Phylum=="Omnitrophica" | Phylum=="Parcubacteria")
ps.b.cpr.glom <- tax_glom(ps.b.cpr, taxrank = 'Genus', NArm = FALSE)
ps.b.cpr.glom.df <- data.table(psmelt(ps.b.cpr.glom))

ps.b.cpr.glom.df$Class <- as.character(ps.b.cpr.glom.df$Class)
ps.b.cpr.glom.df$Phylum <- as.character(ps.b.cpr.glom.df$Phylum)

ps.b.cpr.glom.df[, median := median(Abundance, na.rm = TRUE), 
                        by = "Class"]

ps.b.cpr.glom.df[(median <= 0.001 & Phylum == "Parcubacteria"), Class := "Other Parcubacteria"]
ps.b.cpr.glom.df[(median <= 0.001 & Phylum == "Omnitrophica"), Class := "Other Omnitrophica"]
ps.b.cpr.glom.df$Class[is.na(ps.b.cpr.glom.df$Class) & ps.b.cpr.glom.df$Phylum=="Omnitrophica"] <- "Unc. Omnitrophica"
ps.b.cpr.glom.df$Class[is.na(ps.b.cpr.glom.df$Class) & ps.b.cpr.glom.df$Phylum=="Parcubacteria"] <- "Unc. Parcubacteria"
ps.b.cpr.glom.df <- data.frame(lapply(ps.b.cpr.glom.df, 
                                               function(x) {gsub("Candidatus_", "Cand. ", x)}))
ps.b.cpr.glom.df <- data.frame(lapply(ps.b.cpr.glom.df, 
                                               function(x) {gsub("Omnitrophica_Incertae_Sedis", "Omnitrophica Inc. Sed.", x)}))

names(ps.b.gen.glom.df)[names(ps.b.gen.glom.df) == 'Genus'] <- 'Taxa'
names(ps.b.cpr.glom.df)[names(ps.b.cpr.glom.df) == 'Class'] <- 'Taxa'

ps.b.gen.glom.df.f = ps.b.gen.glom.df[, c("Taxa", "Abundance", "Site.name", "Month", "Time_point")]
ps.b.cpr.glom.df.f = ps.b.cpr.glom.df[, c("Taxa", "Abundance", "Site.name", "Month", "Time_point")]

ps.b.cpr.gen.glom.df <- rbind(ps.b.gen.glom.df.f, ps.b.cpr.glom.df.f)

ps.b.cpr.gen.glom.df$Taxa = factor(ps.b.cpr.gen.glom.df$Taxa, 
                                                       levels = c("Gallionella","Sideroxydans","Ferriphaselus",
                                                                  "Sulfurovum","Sulfuricurvum",
                                                                  "Cand. Azambacteria","Cand. Falkowbacteria","Cand. Jorgensenbacteria",
                                                                  "Cand. Nomurabacteria","Cand. Moranbacteria","Cand. Magasanikbacteria",
                                                                  "Cand. Wolfebacteria","Other Parcubacteria","Unc. Parcubacteria",
                                                                  "Omnitrophica Inc. Sed.","Unc. Omnitrophica"))
ps.b.cpr.gen.glom.df$Abundance = as.numeric(as.character(ps.b.cpr.gen.glom.df$Abundance))

cpr.gen_cols = colorRampPalette(brewer.pal(8, "Accent"))(16)
cpr.gen_cols_c = gsub("#DD2456", "#9c183c", cpr.gen_cols)
cpr.gen_cols_c = gsub("#A0BAAC", "#749a85", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#E4B9A3", "#489fca", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#FEEB93", "#f58e03", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#FDC988", "#c988fd", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#D1DD9E", "#b6c965", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#FDC086", "#fc993b", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#FFFF99", "springgreen4", cpr.gen_cols_c)
cpr.gen_cols_c = gsub("#BEAED4", "violet", cpr.gen_cols_c)

ps.b.cpr.gen.glom.df.plot <- ggplot(ps.b.cpr.gen.glom.df, 
                                     aes(x = Taxa, y = Abundance, fill = Taxa)) + 
  facet_grid(Site.name~Month)+
  geom_hline(yintercept = 0.5,
           color="white",
           linetype="solid") +
  geom_point(aes(color=Taxa, fill=Taxa,
                 size=Abundance), alpha = 0.8) +
  scale_y_sqrt(breaks=c(0.05,0.25,0.75),
               limits = c(0,1),
               labels = scales::percent) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.text.y = element_text(angle = 0),
        strip.placement = "outside",
        strip.background = element_rect(colour = "gray88",
                                        fill = "gray88"),
        panel.background = element_rect(colour = "gray91",
                                       fill = "gray91"),
        panel.grid.major.y = element_line(color = "white"),
        legend.title = element_text(face = "bold")) + 
  guides(fill = guide_legend(nrow = 16),
         shape = NULL,
         size = NULL,
         alpha = NULL) +
  ylab("Relative Abundance\n") +
  scale_fill_manual(values = cpr.gen_cols_c,
                    na.value = "gray48") +
  scale_color_manual(values = cpr.gen_cols_c,
                    na.value = "gray48")
```
```{r echo=FALSE, fig.width=14, message=FALSE, warning=FALSE, fig.height=11}
ps.b.cpr.gen.glom.df.plot
```

Unc. Omnitrophica seems to be correlated with Gallionella abundances. Let's take a closer look!

```{r echo=FALSE, message=FALSE, warning=FALSE}
ps.b.cpr.gen.glom.df.gal = subset(ps.b.cpr.gen.glom.df, 
                                  Taxa=="Gallionella",
                                  select=c("Taxa","Abundance"))
names(ps.b.cpr.gen.glom.df.gal) = c("Gal","Gal_Abund")
ps.b.cpr.gen.glom.df.uncomni = subset(ps.b.cpr.gen.glom.df, 
                                  Taxa=="Unc. Omnitrophica",
                                  select=c("Taxa","Abundance"))
names(ps.b.cpr.gen.glom.df.uncomni) = c("UncOmni","UncOmni_Abund")
ps.b.cpr.gen.gal.uncomni <- cbind(ps.b.cpr.gen.glom.df.gal, ps.b.cpr.gen.glom.df.uncomni)
ps.b.cpr.gen.gal.uncomni$Gal_Abund = as.numeric(as.character(ps.b.cpr.gen.gal.uncomni$Gal_Abund))
ps.b.cpr.gen.gal.uncomni$UncOmni_Abund = as.numeric(as.character(ps.b.cpr.gen.gal.uncomni$UncOmni_Abund))

formula <- y ~ x
gal_r2 = round(summary(lm(Gal_Abund ~ UncOmni_Abund,ps.b.cpr.gen.gal.uncomni))$r.squared, digits = 3)
gal_lm = lm(Gal_Abund ~ UncOmni_Abund,ps.b.cpr.gen.gal.uncomni)
# https://www.r-bloggers.com/can-we-do-better-than-r-squared/
# anova to calculate regalual sum of squares
gal.anova <- anova(gal_lm)
gal.tss <- sum(gal.anova$"Sum Sq")
gal.pr <- residuals(gal_lm)/(1 - lm.influence(gal_lm)$hat)
gal.PRESS <- sum(gal.pr^2)
# predictive R^2
gal.pred.r.squared <- 1 - gal.PRESS/(gal.tss)

ps.b.cpr.gen.gal.uncomni_plot = ggplot(ps.b.cpr.gen.gal.uncomni,
                                       aes(x=Gal_Abund, y=UncOmni_Abund)) +
  geom_point() +
  geom_smooth(method = "lm") +
  stat_cor(label.y = 0.19, label.x = 0.15, 
           size= 7) +
  stat_poly_eq(
    aes(label = ..eq.label..),
    formula = formula, 
    label.y = 0.21, label.x = 0.15,
    parse = TRUE, size= 7) +
  annotate("text", x = 0.15, y = 0.23, 
           label = "paste(italic(R) ^ 2, \" = 0.973\")", 
           parse = TRUE, size= 7) +
  # annotate("text", x = 0.1, y = 0.22, 
  #        label = "paste(Pred.,italic(R) ^ 2, \" = 0.972\")", parse = TRUE) +
  scale_x_continuous(limits = c(0,0.54),
                     expand = c(0,0)) +
  scale_y_continuous(limits = c(0,0.25),
                     expand = c(0,0)) +
  labs(x="Gallionella genus relative abundance (%)",
       y="Unc. Omnitrophica taxa relative abundance (%)")
  

ps.b.cpr.gen.glom.df.sid = subset(ps.b.cpr.gen.glom.df, 
                                  Taxa=="Sideroxydans",
                                  select=c("Taxa","Abundance"))
names(ps.b.cpr.gen.glom.df.sid) = c("sid","sid_Abund")

ps.b.cpr.gen.sid.uncomni <- cbind(ps.b.cpr.gen.glom.df.sid, ps.b.cpr.gen.gal.uncomni)
ps.b.cpr.gen.sid.uncomni$sid_Abund = as.numeric(as.character(ps.b.cpr.gen.sid.uncomni$sid_Abund))
ps.b.cpr.gen.sid.uncomni$UncOmni_Abund = as.numeric(as.character(ps.b.cpr.gen.sid.uncomni$UncOmni_Abund))

sid_r2 = round(summary(lm(sid_Abund ~ UncOmni_Abund,ps.b.cpr.gen.sid.uncomni))$r.squared, 
               digits = 3)
sid_lm = lm(sid_Abund ~ UncOmni_Abund,ps.b.cpr.gen.sid.uncomni)
# https://www.r-bloggers.com/can-we-do-better-than-r-squared/
# anova to calculate residual sum of squares
sid.anova <- anova(sid_lm)
sid.tss <- sum(sid.anova$"Sum Sq")
sid.pr <- residuals(sid_lm)/(1 - lm.influence(sid_lm)$hat)
sid.PRESS <- sum(sid.pr^2)
# predictive R^2
sid.pred.r.squared <- 1 - sid.PRESS/(sid.tss)

ps.b.cpr.gen.sid.uncomni_plot = ggplot(ps.b.cpr.gen.sid.uncomni,
                                       aes(x=sid_Abund, y=UncOmni_Abund)) +
  geom_point() +
  geom_smooth(method = "lm") +
  stat_cor(label.y = 0.19, label.x = 0.1, 
           size= 7) +
  stat_poly_eq(
    aes(label = ..eq.label..),
    formula = formula, 
    label.y = 0.21, label.x = 0.1,
    parse = TRUE, size= 7) +
  annotate("text", x = 0.1, y = 0.23, 
           label = "paste(italic(R) ^ 2, \" = 0.961\")", 
           parse = TRUE, size= 7) +
  # annotate("text", x = 0.1, y = 0.22, 
  #        label = "paste(Pred.,italic(R) ^ 2, \" = 0.957\")", parse = TRUE) +
  scale_x_continuous(limits = c(0,0.42),
                     expand = c(0,0)) +
  scale_y_continuous(limits = c(0,0.25),
                     expand = c(0,0))+
  labs(x="Sideroxydans genus relative abundance (%)",
       y="Unc. Omnitrophica taxa relative abundance (%)")
```
```{r echo=FALSE, fig.width=16, message=FALSE, warning=FALSE}
plot_grid(ps.b.cpr.gen.gal.uncomni_plot, ps.b.cpr.gen.sid.uncomni_plot, 
          ncol = 2, rel_widths = c(1,1))
```

Uh lala! :D

But I have doubts about this. This is basically the result of agglomeration via ```phyloseq::tax_glom``` of the SV table at genus level which may cause some inflation in this kind of stuff. This if everything is correct.

So does it work the same way at SV-level?

```{r echo=FALSE, message=FALSE, warning=FALSE}
ps.b.gal = subset_taxa(ps.b.r, 
                      Genus=="Gallionella")
ps.b.sid = subset_taxa(ps.b.r, 
                      Genus=="Sideroxydans")
ps.b.omni = subset_taxa(ps.b.r, 
                      Phylum=="Omnitrophica" | Class == "NA")

table_ps.b.gal=as.data.frame(otu_table(ps.b.gal))
melt_table_ps.b.gal <- melt(table_ps.b.gal, id.vars=1)
f_melt_table_ps.b.gal = as.data.frame(melt_table_ps.b.gal[, c("value")])
colnames(f_melt_table_ps.b.gal) = "Gallionella"

table_ps.b.sid=as.data.frame(otu_table(ps.b.sid))
melt_table_ps.b.sid <- melt(table_ps.b.sid, id.vars=1)
f_melt_table_ps.b.sid = as.data.frame(melt_table_ps.b.sid[, c("value")])
colnames(f_melt_table_ps.b.sid) = "Sideroxydans"

table_ps.b.omni=as.data.frame(otu_table(ps.b.omni))
melt_table_ps.b.omni <- melt(table_ps.b.omni, id.vars=1)
f_melt_table_ps.b.omni = as.data.frame(melt_table_ps.b.omni[, c("value")])
colnames(f_melt_table_ps.b.omni) = "Omnitrophica"

#subsample the thing based on Sideroxydans

ss_f_melt_table_ps.b.gal=as.data.frame(f_melt_table_ps.b.gal[sample(nrow(f_melt_table_ps.b.gal), 28665), ])
colnames(ss_f_melt_table_ps.b.gal) = "Gallionella"
ss_f_melt_table_ps.b.omni=as.data.frame(f_melt_table_ps.b.omni[sample(nrow(f_melt_table_ps.b.omni), 28665), ])
colnames(ss_f_melt_table_ps.b.omni) = "Omnitrophica"

ps.b.propercorrs = cbind(ss_f_melt_table_ps.b.gal,
                         f_melt_table_ps.b.sid,
                         ss_f_melt_table_ps.b.omni)

ps.b.propercorrs.gal.uncomni_plot = ggplot(ps.b.propercorrs,
                                       aes(x=Gallionella, y=Omnitrophica)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_x_log10() +
  scale_y_log10() +
  labs(x="Gallionella genus relative abundance (%)",
       y="Unc. Omnitrophica taxa relative abundance (%)")
  
ps.b.propercorrs.sid.uncomni_plot = ggplot(ps.b.propercorrs,
                                       aes(x=Sideroxydans, y=Omnitrophica)) +
  geom_point() +
  geom_smooth(method = "lm") +
  scale_x_log10() +
  scale_y_log10() +
  labs(x="Sideroxydans genus relative abundance (%)",
       y="Unc. Omnitrophica taxa relative abundance (%)")
```
```{r echo=FALSE, fig.width=16, message=FALSE, warning=FALSE}
plot_grid(ps.b.propercorrs.gal.uncomni_plot, ps.b.propercorrs.sid.uncomni_plot, 
          ncol = 2, rel_widths = c(1,1))
```

So guess what, it was nothing! Bah. SV-level plots show little or no correlation between relative abundances of these taxa.