---
title: "SGG - What explains proteobacterial co-dynamics in the SGG dataset?"
author: "André Soares"
date: "8 January 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r initial loadings, message=FALSE, warning=FALSE, include=FALSE}
library(dada2); packageVersion("dada2")
library(ggplot2); packageVersion("ggplot2")
library(phyloseq); packageVersion("phyloseq")
library(data.table); packageVersion("data.table")
library(DESeq2); packageVersion("DESeq2")
library(dplyr)
library(plyr)
library(magrittr)
library(RColorBrewer)
library(phangorn)
library(DECIPHER)
library("structSSI")
library(ips)
library(msa)
library(picante)
library(igraph)
library(cowplot)
library(plotly)
library(lemon)
library(GGally)
library(randomcoloR)
library(knitr)
library(kableExtra)
library(ggpmisc)
library(ggpubr)
```

```{r load SV data, echo=FALSE}
SGG_SVt = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/seqtab_final.rds")
SGG_Tax = readRDS("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_data_trimmed_DADA2_table/tax_final.rds")
sgg_metadata<-read.csv("/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/SGG_16S_metadata/SGG_metadata.csv", header=TRUE)
order<-rownames(SGG_SVt)
sgg_metadata_ord<-sgg_metadata[match(order, sgg_metadata$Sample_Code),]
rownames(sgg_metadata_ord) <- sgg_metadata_ord$Sample_Code
ps <- phyloseq(otu_table(SGG_SVt, taxa_are_rows=FALSE),
               sample_data(sgg_metadata_ord), 
               tax_table(SGG_Tax))
ps.a = subset_samples(ps, Site.name != "Taffs Well PUMPED")
ps.b = subset_samples(ps.a, Sample_type != "Control")
ps.b <- prune_taxa(taxa_sums(ps.b) > 0, ps.b)
#order months
sample_data(ps.b)$Month = factor(sample_data(ps.b)$Month, 
                                 levels = c("April","August","December"))
#to relative abundances
ps.b = subset_taxa(ps.b, Kingdom %in% c("Archaea", "Bacteria"))
ps.b.r <-  transform_sample_counts(ps.b, function(x) {x/sum(x)} ) 
```

##**1 -** High-level taxonomic view of the dataset:

```{r general proteobacterial taxonomy, echo=FALSE}
ps.b.r.glom <- tax_glom(ps.b.r, taxrank = 'Phylum', NArm = FALSE)
ps.b.r.glom.psdf <- data.table(psmelt(ps.b.r.glom))
ps.b.r.glom.psdf$Phylum <- as.character(ps.b.r.glom.psdf$Phylum)

ps.b.r.glom.psdf[, median := median(Abundance, na.rm = TRUE), 
                 by = "Phylum"]
ps.b.r.glom.psdf[(median <= 0.01), Phylum := "Other"]
#Removing Proteobacteria from previous table
ps.b.r.glom.psdf.noProteo<-ps.b.r.glom.psdf[!grepl("Proteobacteria", ps.b.r.glom.psdf$Phylum),]
#New table, with Proteobacterial classes only
ps.b.r.ProtCl = subset_taxa(ps.b.r, Phylum == "Proteobacteria")
ps.b.r.ProtCl.glom <- tax_glom(ps.b.r.ProtCl, taxrank = 'Class', NArm = FALSE)
ps.b.r.ProtCl.glom.psdf <- data.table(psmelt(ps.b.r.ProtCl.glom))
ps.b.r.ProtCl.glom.psdf$Class <- as.character(ps.b.r.ProtCl.glom.psdf$Class)

ps.b.r.ProtCl.glom.psdf[, median := median(Abundance, na.rm = TRUE), 
                        by = "Class"]
ps.b.r.ProtCl.glom.psdf[(median <= 0.01), Class := "Other Proteobacteria"]
ps.b.r.ProtCl.glom.psdf.noP <- subset(ps.b.r.ProtCl.glom.psdf, select = -Phylum)

names(ps.b.r.glom.psdf.noProteo)[names(ps.b.r.glom.psdf.noProteo) == 'Phylum'] <- 'Taxa'
names(ps.b.r.ProtCl.glom.psdf.noP)[names(ps.b.r.ProtCl.glom.psdf.noP) == 'Class'] <- 'Taxa'
ps.b.r.ProteoCl.AllPhy <- rbind(ps.b.r.glom.psdf.noProteo, ps.b.r.ProtCl.glom.psdf.noP)

ps.b.r.ProteoCl.AllPhy$Site.name = factor(ps.b.r.ProteoCl.AllPhy$Site.name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
ps.b.r.ProteoCl.AllPhy$Taxa = factor(ps.b.r.ProteoCl.AllPhy$Taxa, 
                                                       levels = c("Betaproteobacteria","Deltaproteobacteria",
                                                                  "Epsilonproteobacteria","Gammaproteobacteria","Other Proteobacteria",
                                                                  "Actinobacteria","Nitrospirae","Omnitrophica","Parcubacteria",
                                                                  "Other"))
ps.b.r.ProteoCl.AllPhy.plot<-ggplot(ps.b.r.ProteoCl.AllPhy[Abundance > 0], 
                                    aes(x = Site.name, y = Abundance/3, fill = Taxa)) + 
  facet_grid(Month~.) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = brewer.pal(10, "Paired"),
                     na.value = "gray40") +
  scale_y_continuous(labels = scales::percent) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(angle = 30, hjust = 1)) + 
  guides(fill = guide_legend(keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance\n")
```
```{r echo=FALSE, fig.height=14, fig.width=20}
ps.b.r.ProteoCl.AllPhy.plot
```

At a first glance, a dataset dominated by Epsilonproteobacteria and Betaproteobacteria, with some CPR phyla (Omnitrophica and Parcubacteria) presence. Sites that smelt like sulphur during sampling are populated by Epsilonproteobacteria, while iron and iron-hydroxide filled sites have strong Betaproteobacterial presence on most occasions, showing Omnitrophica and Parcubacteria occupying close to 30-40% of the taxonomic profiles. Lindsay stands alone, showing CPR phyla occupying most of the profile and low-abundance but numerous phyla representing close to 40% of the profile throughout the year.

But is this seen at lower taxonomic levels?

##**2 -** Genus-level view of the dataset:

```{r genera plot, echo=FALSE}
ps.b.deep_128 = subset_taxa(ps.b.r, 
                            Family=="Gallionellaceae" | Family=="Helicobacteraceae" | Family=="Pseudomonadaceae" | Family=="Thiotrichaceae" | Family=="Hydrogenophilaceae" | Phylum=="Omnitrophica" | Phylum=="Parcubacteria")
ps.b.deep_128.glom <- tax_glom(ps.b.deep_128, taxrank = 'Genus', NArm = FALSE)
ps.b.deep_128.glom.df <- data.table(psmelt(ps.b.deep_128.glom))
levels(ps.b.deep_128.glom.df$Genus)<-c(levels(ps.b.deep_128.glom.df$Genus),
                                     c("Unc. Gallionellaceae","Unc. Helicobacteraceae", 
                                       "Omnitrophica","Parcubacteria","Unc. Omnitrophica","Unc. Parcubacteria",
                                       "Unc. Pseudomonadaceae","Unc. Thiotrichaceae","Unc. Hydrogenophilaceae"))
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Gallionellaceae"] <- "Unc. Gallionellaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Helicobacteraceae"] <- "Unc. Helicobacteraceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Pseudomonadaceae"] <- "Unc. Pseudomonadaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Thiotrichaceae"] <- "Unc. Thiotrichaceae"
ps.b.deep_128.glom.df$Genus[is.na(ps.b.deep_128.glom.df$Genus) & ps.b.deep_128.glom.df$Family=="Hydrogenophilaceae"] <- "Unc. Hydrogenophilaceae"
ps.b.deep_128.glom.df$Genus[ps.b.deep_128.glom.df$Phylum=="Omnitrophica"] <- "Omnitrophica"
ps.b.deep_128.glom.df$Genus[ps.b.deep_128.glom.df$Phylum=="Parcubacteria"] <- "Parcubacteria"
ps.b.deep_128.glom.df$Site.name = factor(ps.b.deep_128.glom.df$Site.name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
ps.b.r.glom.psdf.unk.glom <- tax_glom(ps.b.r, taxrank = 'Genus', NArm = FALSE)
ps.b.r.glom.psdf.unk <- data.table(psmelt(ps.b.r.glom.psdf.unk.glom))
levels(ps.b.r.glom.psdf.unk$Phylum)<-c(levels(ps.b.r.glom.psdf.unk$Phylum),
                                     c("Unknown Phyla"))
levels(ps.b.r.glom.psdf.unk$Genus)<-c(levels(ps.b.r.glom.psdf.unk$Genus),
                                     c("Unknown Phyla"))
ps.b.r.glom.psdf.unk$Phylum[is.na(ps.b.r.glom.psdf.unk$Phylum)]="Unknown Phyla"
ps.b.r.glom.psdf.unk$Genus[ ps.b.r.glom.psdf.unk$Phylum=="Unknown Phyla"] <- "Unknown Phyla"
ps.b.r.glom.psdf.unkonly<-subset(ps.b.r.glom.psdf.unk, 
                                                  Phylum %in% c("Unknown Phyla"))
ps.b.deep_128.glom.df.unkonly <- rbind(ps.b.deep_128.glom.df, ps.b.r.glom.psdf.unkonly)

gen_cols = colorRampPalette(brewer.pal(8, "Accent"))(15)
gen_cols_c = gsub("#DD2456", "#9c183c", gen_cols)
gen_cols_c = gsub("#A0BAAC", "#749a85", gen_cols_c)
gen_cols_c = gsub("#E4B9A3", "#489fca", gen_cols_c)
gen_cols_c = gsub("#FEEB93", "#f58e03", gen_cols_c)
gen_cols_c = gsub("#FDC988", "#c988fd", gen_cols_c)
gen_cols_c = gsub("#D1DD9E", "#b6c965", gen_cols_c)
gen_cols_c = gsub("#FDC086", "#fc993b", gen_cols_c)
gen_cols_c = gsub("#FFFF99", "springgreen4", gen_cols_c)
gen_cols_c = gsub("#BEAED4", "violet", gen_cols_c)

ps.b.deep_128.glom.df.unkonly.f<-subset(ps.b.deep_128.glom.df.unkonly, 
                                                  Abundance > 0.03)
ps.b.deep_128.glom.df.unkonly.f$Genus = factor(ps.b.deep_128.glom.df.unkonly.f$Genus, 
                                                       levels = c("Gallionella","Sideroxydans","Ferriphaselus","Unc. Gallionellaceae",
                                                                  "Sulfurovum","Sulfurimonas","Sulfuricurvum","Sulfuricella","Sulfuriferula",
                                                                  "Pseudomonas","Thiothrix","Unc. Hydrogenophilaceae",
                                                                  "Parcubacteria","Omnitrophica",
                                                                  "Unknown Phyla"))

ps.b.deep_128.glom.df.plot <- ggplot(ps.b.deep_128.glom.df.unkonly.f, 
                                     aes(x = Genus, y = Abundance, fill = Genus)) + 
  facet_grid(Site.name~Month)+
  geom_hline(yintercept = 0.5,
           color="white",
           linetype="solid") +
  geom_point(aes(color=Genus, fill=Genus,
                 size=Abundance), alpha = 0.8) +
  scale_y_continuous(breaks = c(0.25,0.75),
                     limits = c(0,1) ) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.text.y = element_text(angle = 0),
        strip.placement = "outside",
        strip.background = element_rect(colour = "gray88",
                                        fill = "gray88"),
        panel.background = element_rect(colour = "gray91",
                                       fill = "gray91"),
        panel.grid.major.y = element_line(color = "white"),
        legend.title = element_text(face = "bold")) + 
  guides(fill = guide_legend(nrow = 16),
         shape = NULL,
         size = NULL,
         alpha = NULL) +
  ylab("Relative Abundance\n") +
  scale_fill_manual(values = gen_cols_c,
                    na.value = "gray48") +
  scale_color_manual(values = gen_cols_c,
                    na.value = "gray48")
```
```{r echo=FALSE, fig.width=14, message=FALSE, warning=FALSE, fig.height=11}
ps.b.deep_128.glom.df.plot
```

It seems that things are not that simple in this dataset! To put it simply, Epsilonproteobacteria-rich samples are co-dominated by Sulfuricurvum and Sulfurovum and Betaproteobacteria-rich samples by Gallionella and Sideroxydans. Lindsay, as seen before, is mostly made up of low-abundance taxa (not shown) with CPR phyla representing close to 50% of the profile across the year. Taff's well, albeit iron-rich, probably due to its external influences being on the edge of the coalfield, is populated by a mix of iron- and sulfur-oxidisers in April, which then turns into a FeOB-dominated profile in August, with Ferriphaselus going up to 50% in December.

It's unclear why bacteria with very similar metabolisms (iron- and sulfur-oxidisers) would co-occur and seemingly compete for dominance of the profiles this way. What could drive this? What do available genomes for Sulfuricurvum, Sulfurovum, Gallionella, Sideroxydans and Ferriphaselus say?

##**3 -** Genomic information for the most abundant genera in the dataset:

#### Gallionella:

Gallionella is an microaerophilic iron-oxidiser that uses $Fe^{+2}$ solely as electron donor. It's equipped for aerotaxis and motile. Grows preferrably at lower temperatures (6ºC) and is unable to grow at 30ºC. Carbon metabolism works through the RubisCO pathway.

```{r echo=FALSE}
gal_path = "/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Gallionella_Sideroxydans_genomes/Gallionella_genome_Emerson_2013.png"
include_graphics(gal_path)
```

#### Sideroxydans:

Sideroxydans is a motile microaerophilic iron-oxidiser that uses $Fe^{+2}$ as electron donor but also has the potential to use thiosulfate. This is due to the presence of a *sox* operon (ABXYZ) in its genome. Grows preferrably at higher temperatures (30ºC) and is unable to grow at 6ºC. Carbon metabolism works through the RubisCO pathway.

```{r echo=FALSE}
sid_path ="/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Gallionella_Sideroxydans_genomes/Sideroxydans_genome_Emerson_2013.png"
include_graphics(sid_path)
```

#### Ferriphaselus:

Ferriphaselus is a motile microaerophilic iron-oxidiser that uses $Fe^{+2}$ solely as electron donor. However, it lacks the *Mto* gene cluster to oxidise $Fe^{+2}$ and seems to use instead a panoply of alternative FeO metabolisms specially including *act* (AB1B2CDEF). Further studies are needed to prove its capacity to grow on sulfur species, since not only it has a *sox* operon (ABXYZ) in its genome but also *dsr* (ABEFHCMKJOPN), *rdsr* (DAGC), *Sqr* and *soe* (ABC). Grows preferrably at higher temperatures (30ºC) and is unable to grow at 6ºC. Carbon metabolism works through the RubisCO pathway.

#### Sulfuricurvum:

Sulfuricurvum is a motile facultative anaerobe that can use sulfide, sulfite and elemental sulfur as electron donors along with oxygen, nitrate and nitric oxide as electron acceptors. To allow that number of electron donors, Sulfuricurvum not only uses the *sox* (ABXYZ) operon, but also *Sqr* (DF), *Sor* (AB) and *Fcc* (AB). 4 plasmids were found in Handley *et al.* (2014) with one associated to nitrogen metabolism. Carbon metabolism works through the rTCA cycle.

```{r echo=FALSE}
sid_path ="/media/andre/B2F8C9A0F8C962E9/SGG_16S_analysis/Sulfurovum_Sulfuricurvum_genomes/Sulfuricurvum_genome_Handley_2014.png"
include_graphics(sid_path)
```

#### Sulfurovum:

Sulfurovum is a non-motile facultative anaerobe that can use thiosulfate and elemental sulfur as electron donors along with oxygen and nitrate  as electron acceptors. Sulfurovum only uses the *sox* (ABXYZ) operon, as opposed to Sulfuricurvum. Carbon metabolism works through the rTCA cycle.

### Genomic hits for FeOB metal resistance potential:

**Genome Accessions:** NC_014394.1, NC_013959.1, BBTH01000001:BBTH01000023, NZ_CP011308.1, CP003920.1.

Genome annotation done with ```PROKKA v1.12``` (https://github.com/tseemann/prokka), functional annotation with ```EggNOG-mapper``` (http://eggnogdb.embl.de/app/emapper#/app/emapper).

```{r echo=FALSE}
Gal = c(3,3,14,3,1,1,1,3,0,15)
Sid = c(0,2,4,2,5,1,1,2,1,5)
Fer = c(0,4,6,2,2,0,2,2,2,7)
Sulfuric = c(0,3,4,2,3,1,0,2,0,2)
Sulfurov = c(0,3,5,4,6,1,2,3,1,3)
Els = c("Mercury","Arsenic","Cobalt","Zinc","Copper","Manganese","Selenium","Mg/Co Transport","Magnesium","Heavy metal efflux pump/-related
")

g_hits_df = data.frame("Metals" = Els, 
                       "Gallionella" = Gal, 
                       "Sideroxydans" = Sid, 
                       "Ferriphaselus" = Fer,
                       "Sulfuricurvum" = Sulfuric,
                       "Sulfurovum" = Sulfurov)
options(knitr.table.format = "html") 

g_hits_df %>%
mutate_if(is.numeric, function(x) {
cell_spec(x, "html", bold = T, color = spec_color(x, end = 0.9),
          font_size = spec_font_size(x))
  }) %>%
kable("html", escape = F, align = "c") %>%
  kable_styling(bootstrap_options = c("striped"),
                full_width = F)
```

As advertised, *Gallionella* seems to be a lot more prepared to thrive in heavy-metal-rich environments. Maybe that gives this genus an advantage in colonizing most of the iron-rich sites in South Wales. Not a lot can be taken from this since these genomes were generated from far away sites across the world. They do, however, reveal the potential in some of these genera for competitive advantages in heavy-metal-rich sites.

Ok then, so do any of the metals we measured have any impact on the relative abundance of any of these genera?

##**4 -** Metals and relative abundance of important genera:

```{r echo=FALSE}
metadata_cor_main_genera = ps.b.deep_128.glom.df[, c("Genus", "Abundance", "Site.name", "Month", "Time_point","SO4..µg.ml",
                                                     "DO..sat","Temp.C","pH",
                                                     "X55.Mn.ng.ml","Mg.µg.ml","X68.Zn.ng.ml","Na.µg.ml","X6.Li.ng.ml",
                                                     "X135.Ba.ng.ml","X11.B.ng.ml","X75.As..He..ng.ml","X59.Co.ng.ml",
                                                     "X63.Cu.ng.ml","X78.Se.ng.ml","X200.Hg.ng.ml","X201.Hg.ng.ml","X202.Hg.ng.ml")]
names(metadata_cor_main_genera) = c("Genus", "Abundance", "Site_name", "Month", 
                                    "Time_point","SO4","DO","Temp","pH"
                                    ,"Mn","Mg","Zn","Na","Li","Ba",
                                    "B","As","Co","Cu","Se","X200_Hg","X201_Hg","X202_Hg")
#medium
#removed zinc and mg because they didn't show any relevant patterns
medium_metadata_cor_main_genera = metadata_cor_main_genera[, c("Genus", "Abundance", "Site_name", "Month", "Time_point",
                                                               "Na","Li","Ba","B")]
medium_metadata_cor_main_genera=medium_metadata_cor_main_genera[complete.cases(medium_metadata_cor_main_genera$Genus), ]
dm1 <- melt(medium_metadata_cor_main_genera[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
dm1 = dm1[, c("Site_name", "Month","value","Abundance")]
colnames(dm1) <- c("Site_name", "Month", "Genus", "Abundance")
dm2 <- melt(medium_metadata_cor_main_genera[,c("Site_name","Month","Na","Li","Ba","B")], id=c("Site_name","Month"))
colnames(dm2) <- c("Site_name", "Month", "metals", "metal_ppm")
medium_metadata_cor_main_genera_melt <- merge(dm1, dm2, allow.cartesian=TRUE)

medium_metadata_cor_main_genera_melt$Site_name = factor(medium_metadata_cor_main_genera_melt$Site_name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
medium_metadata_cor_main_genera_melt$Genus = factor(medium_metadata_cor_main_genera_melt$Genus, 
                                                       levels = c("Gallionella","Sideroxydans","Ferriphaselus","Unc. Gallionellaceae",
                                                                  "Sulfurovum","Sulfurimonas", "Unc. Helicobacteraceae",
                                                                  "Sulfuricurvum","Sulfuriferula"))
medium_metadata_cor_main_genera_melt_plot<-ggplot(medium_metadata_cor_main_genera_melt, 
                        aes(x=Site_name)) +
  geom_point(aes(y=Abundance,
                 fill=Genus,
                 shape=Month,
                 size=Abundance,
                 alpha=0.5),
             color="transparent",
             position = position_dodge(width=0.5)) +
  geom_point(aes(y=metal_ppm/400,
                 shape=Month,
                 color=metals),
             size = 4,
             position = position_dodge(width=0.5)) +
  scale_fill_manual(values = colorRampPalette(brewer.pal(8, "Dark2"))(10)) +
  scale_colour_manual(values = brewer.pal(6, "Paired")) +
  scale_shape_manual(values = c(21,22,23)) +
  scale_y_continuous(sec.axis = sec_axis(~.*400, name = "Concentrations in ppm")) +
  labs(y="Relative Abundance (%)") +
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(size=14),
        axis.text.x = element_text(angle=30, hjust = 1,vjust = 1),
        axis.text.y = element_text(size=12),
        legend.position = "right",
        legend.justification = "center",
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13, face = "bold"),
        legend.direction = "vertical") +
  guides(fill = guide_legend(override.aes = list(colour=colorRampPalette(brewer.pal(8, "Dark2"))(10),
                                                 shape=15, size=3)),
         size=NULL)
#low-medium
low_medium_metadata_cor_main_genera = metadata_cor_main_genera[, c("Genus", "Abundance", "Site_name", "Month", "Time_point",
                                                               "As","Co","Cu","Se")]
low_medium_metadata_cor_main_genera=low_medium_metadata_cor_main_genera[complete.cases(low_medium_metadata_cor_main_genera$Genus), ]
dm1 <- melt(low_medium_metadata_cor_main_genera[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
dm1 = dm1[, c("Site_name", "Month","value","Abundance")]
colnames(dm1) <- c("Site_name", "Month", "Genus", "Abundance")
dm2 <- melt(low_medium_metadata_cor_main_genera[,c("Site_name","Month","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(dm2) <- c("Site_name", "Month", "metals", "metal_ppm")
low_medium_metadata_cor_main_genera_melt <- merge(dm1, dm2, allow.cartesian=TRUE)

low_medium_metadata_cor_main_genera_melt$Site_name = factor(low_medium_metadata_cor_main_genera_melt$Site_name, 
                                                       levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
low_medium_metadata_cor_main_genera_melt$Genus = factor(low_medium_metadata_cor_main_genera_melt$Genus, 
                                                       levels = c("Gallionella","Sideroxydans","Ferriphaselus","Unc. Gallionellaceae",
                                                                  "Sulfurovum","Sulfurimonas", "Unc. Helicobacteraceae",
                                                                  "Sulfuricurvum","Sulfuriferula"))
low_medium_metadata_cor_main_genera_melt_plot<-ggplot(low_medium_metadata_cor_main_genera_melt, 
                        aes(x=Site_name)) +
  geom_point(aes(y=Abundance,
                 fill=Genus,
                 shape=Month,
                 size=Abundance,
                 alpha=0.5),
             color="transparent",
             position = position_dodge(width=0.5)) +
  geom_point(aes(y=metal_ppm/50,
                 shape=Month,
                 color=metals),
             size=4,
             position = position_dodge(width=0.5)) +
  scale_fill_manual(values = colorRampPalette(brewer.pal(8, "Dark2"))(10)) +
  scale_colour_manual(values = brewer.pal(6, "Paired")) +
  scale_shape_manual(values = c(21,22,23)) +
  scale_y_continuous(sec.axis = sec_axis(~.*50, name = "Concentrations in ppm")) +
  labs(y="Relative Abundance (%)") +
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(size=14),
        axis.text.x = element_text(angle=30, hjust = 1,vjust = 1),
        axis.text.y = element_text(size=12),
        legend.position = "right",
        legend.justification = "center",
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 13, face = "bold"),
        legend.direction = "vertical") +
  guides(fill = guide_legend(override.aes = list(colour=colorRampPalette(brewer.pal(8, "Dark2"))(10),
                                                 shape=15, size=3)),
         size=NULL)
```

Not showing high and low concentration value metals because they don't seem to be relevant.

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width=16, fig.height=8}
medium_metadata_cor_main_genera_melt_plot
```

Actually quite interesting, it seems barium is only high in the sulfur sites and in Taffs! Could barium concentrations be driving Ferriphaselus abundances in the dataset? Barium concentrations are also quite different in the three sulfur sites. They seem to drive the Sulfurovum peaks in Crumlin, where as in Six Bells, Na, Li and B are quite high but it seems the presence of barium kind of seems to unlock the dominance of this genus. Much lower concentrations of Na, Li and B with Ba at ~150ppm sends Sulfuricurvum to the dominant position.

It seems higher Na concentrations are related to sites lacking dominance of iron-oxidizers. Also, Ynysarwed has unusual lithium concentrations.

```{r, echo=FALSE, message=FALSE, warning=FALSE, fig.width=16, fig.height=8}
low_medium_metadata_cor_main_genera_melt_plot
```

~50ppm As + ~100ppm Li may explain lack of Gallionellaceae dominance in Ynysarwed. 

But are these the only factors? Here's some more metadata:

```{r main metadata, echo=FALSE}
sample_data(ps.b.r)$Site.name = factor(sample_data(ps.b.r)$Site.name, 
                                          levels = c("Celynen North","Crumlin Navigation","Six Bells",
                                                                  "Blaenavon","Cefn Hengoed","Dinas","Glyncastle",
                                                                  "Morlais","Mountain Gate","Taff Bargoed","Ynysarwed","Lindsay",
                                                                  "Taffs Well"))
metadata<-sample_data(ps.b.r)
#Temp
temp_table.plot<-ggplot(metadata, 
                        aes(x=Site.name, y=Temp.C, fill=Month)) +
  geom_point(aes(fill=Month, color=Month),
             size=2,
             position = position_dodge(width=0.5)) +
  geom_hline(yintercept = 13.4,
             color="dark red",
             linetype="dashed") +
  geom_hline(yintercept = 11,
             color="gray",
             linetype="dashed") +
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(size=10),
        axis.text.x = element_text(angle=30, hjust = 1,vjust = 1),
        axis.text.y = element_text(size=12),
        legend.position = "bottom",
        legend.title = element_text(face = "bold"),
        legend.background = element_rect(fill="gray95", size=.5, linetype="dotted")) +
  annotate("text", "Morlais", 14, vjust = -1, hjust=.45, size= 4,
           label = "Average mine water temperature (Farr et al, 2016)") +
  labs(y=expression(paste("Temperature (",degree,"C)")))
#DO
do.plot<-ggplot(metadata, 
                        aes(x=Site.name, y=`DO..sat`, fill=Month)) +
  geom_point(aes(fill=Month, color=Month),
             size=2,
             position = position_dodge(width=0.5)) +
  geom_hline(yintercept = 50,
           color="gray60",
           linetype="dashed") +
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(size=10),
        axis.text.x = element_text(angle=30, hjust = 1,vjust = 1),
        axis.text.y = element_text(size=12),
        legend.position = "none") +
  labs(y="Dissolved Oxygen (%)")
#pH
pH.plot<-ggplot(metadata, 
                        aes(x=Site.name, y=`pH`, fill=Month)) +
  geom_point(aes(fill=Month, color=Month),
             size=2,
             position = position_dodge(width=0.5)) +
  geom_hline(yintercept = 7,
           color="gray60",
           linetype="dashed") +
  theme(axis.title.x=element_blank(),
        axis.title.y = element_text(size=10),
        axis.text.x = element_text(angle=30, hjust = 1,vjust = 1),
        axis.text.y = element_text(size=12),
        legend.position = "none") +
  labs(y="pH") +
  guides(size=FALSE)
```
```{r  echo=FALSE, message=FALSE, warning=FALSE, fig.height=8, fig.width=10}
plot_grid(temp_table.plot, do.plot, pH.plot, ncol = 1, align='v', rel_heights = c(1,0.8,0.9))
```

Can we try and get numbers out of these supposed relations?

```{r echo=FALSE, message=FALSE, warning=FALSE}
cor_genera_geochem = as(metadata_cor_main_genera[, c("Genus", "Abundance", "Site_name", "Month",
                                                  "Temp","DO","pH","Na","Li","Ba","B","As","Co","Cu","Se")], "data.frame")
cor_genera_geochem_gal = subset(cor_genera_geochem, 
                            Genus=="Gallionella")
cor_genera_geochem_sid = subset(cor_genera_geochem, 
                            Genus=="Sideroxydans")
cor_genera_geochem_fer = subset(cor_genera_geochem, 
                            Genus=="Ferriphaselus")
cor_genera_geochem_sulfuri = subset(cor_genera_geochem, 
                            Genus=="Sulfuricurvum")
cor_genera_geochem_sulfuro = subset(cor_genera_geochem, 
                            Genus=="Sulfurovum")
cor_genera_geochem_gal$Genus = factor(cor_genera_geochem_gal$Genus, 
                                          levels = c("Gallionella"))

cor_genera_geochem_galdm1 <- melt(cor_genera_geochem_gal[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
cor_genera_geochem_galdm1 = cor_genera_geochem_galdm1[, c("Site_name", "Month","value","Abundance")]
colnames(cor_genera_geochem_galdm1) <- c("Site_name", "Month", "Genus", "Abundance")

cor_genera_geochem_galdm2 <- melt(cor_genera_geochem_gal[,c("Site_name","Month","Temp","DO","pH","Na","Li",
                                                            "Ba","B","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(cor_genera_geochem_galdm2) <- c("Site_name", "Month", "variable", "value")

cor_genera_geochem_galdm1dm2_melt <- merge(cor_genera_geochem_galdm1, cor_genera_geochem_galdm2, allow.cartesian=TRUE)

cor_genera_geochem_galdm1dm2_melt_plot=ggplot(cor_genera_geochem_galdm1dm2_melt) +
  geom_jitter(aes(Abundance,value, 
                  colour=variable)) + 
  geom_smooth(aes(Abundance,value, colour=variable), 
              method=lm, se=FALSE) +
  facet_wrap(~variable, scales="free_x") +
  stat_cor(aes(Abundance,value,
               color = variable), 
           label.y = 370, label.x = 0.3)+
  stat_poly_eq(aes(Abundance,value,
        color = variable, label = ..eq.label..),
    formula = y ~ x, 
    label.y = 300, label.x = 0.3,
    parse = TRUE)

cor_genera_geochem_siddm1 <- melt(cor_genera_geochem_sid[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
cor_genera_geochem_siddm1 = cor_genera_geochem_siddm1[, c("Site_name", "Month","value","Abundance")]
colnames(cor_genera_geochem_siddm1) <- c("Site_name", "Month", "Genus", "Abundance")

cor_genera_geochem_siddm2 <- melt(cor_genera_geochem_sid[,c("Site_name","Month","Temp","DO","pH","Na","Li",
                                                            "Ba","B","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(cor_genera_geochem_siddm2) <- c("Site_name", "Month", "variable", "value")

cor_genera_geochem_siddm1dm2_melt <- merge(cor_genera_geochem_siddm1, cor_genera_geochem_siddm2, allow.cartesian=TRUE)

cor_genera_geochem_siddm1dm2_melt_plot=ggplot(cor_genera_geochem_siddm1dm2_melt) +
  geom_jitter(aes(Abundance,value, 
                  colour=variable)) + 
  geom_smooth(aes(Abundance,value, colour=variable), 
              method=lm, se=FALSE) +
  facet_wrap(~variable, scales="free_x") +
  stat_cor(aes(Abundance,value,
               color = variable), 
           label.y = 370, label.x = 0.3)+
  stat_poly_eq(aes(Abundance,value,
        color = variable, label = ..eq.label..),
    formula = y ~ x, 
    label.y = 300, label.x = 0.3,
    parse = TRUE)

cor_genera_geochem_ferdm1 <- melt(cor_genera_geochem_fer[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
cor_genera_geochem_ferdm1 = cor_genera_geochem_ferdm1[, c("Site_name", "Month","value","Abundance")]
colnames(cor_genera_geochem_ferdm1) <- c("Site_name", "Month", "Genus", "Abundance")

cor_genera_geochem_ferdm2 <- melt(cor_genera_geochem_fer[,c("Site_name","Month","Temp","DO","pH","Na","Li",
                                                            "Ba","B","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(cor_genera_geochem_ferdm2) <- c("Site_name", "Month", "variable", "value")

cor_genera_geochem_ferdm1dm2_melt <- merge(cor_genera_geochem_ferdm1, cor_genera_geochem_ferdm2, allow.cartesian=TRUE)

cor_genera_geochem_ferdm1dm2_melt_plot=ggplot(cor_genera_geochem_ferdm1dm2_melt) +
  geom_jitter(aes(Abundance,value, 
                  colour=variable)) + 
  geom_smooth(aes(Abundance,value, colour=variable), 
              method=lm, se=FALSE) +
  facet_wrap(~variable, scales="free_x") +
  stat_cor(aes(Abundance,value,
               color = variable), 
           label.y = 370, label.x = 0.3)+
  stat_poly_eq(aes(Abundance,value,
        color = variable, label = ..eq.label..),
    formula = y ~ x, 
    label.y = 300, label.x = 0.3,
    parse = TRUE)

cor_genera_geochem_sulfuridm1 <- melt(cor_genera_geochem_sulfuri[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
cor_genera_geochem_sulfuridm1 = cor_genera_geochem_sulfuridm1[, c("Site_name", "Month","value","Abundance")]
colnames(cor_genera_geochem_sulfuridm1) <- c("Site_name", "Month", "Genus", "Abundance")

cor_genera_geochem_sulfuridm2 <- melt(cor_genera_geochem_sulfuri[,c("Site_name","Month","Temp","DO","pH","Na","Li",
                                                            "Ba","B","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(cor_genera_geochem_sulfuridm2) <- c("Site_name", "Month", "variable", "value")

cor_genera_geochem_sulfuridm1dm2_melt <- merge(cor_genera_geochem_sulfuridm1, cor_genera_geochem_sulfuridm2, allow.cartesian=TRUE)

cor_genera_geochem_sulfuridm1dm2_melt_plot=ggplot(cor_genera_geochem_sulfuridm1dm2_melt) +
  geom_jitter(aes(Abundance,value, 
                  colour=variable)) + 
  geom_smooth(aes(Abundance,value, colour=variable), 
              method=lm, se=FALSE) +
  facet_wrap(~variable, scales="free_x") +
  stat_cor(aes(Abundance,value,
               color = variable), 
           label.y = 370, label.x = 0.3)+
  stat_poly_eq(aes(Abundance,value,
        color = variable, label = ..eq.label..),
    formula = y ~ x, 
    label.y = 300, label.x = 0.3,
    parse = TRUE)

cor_genera_geochem_sulfurodm1 <- melt(cor_genera_geochem_sulfuro[,c("Site_name","Month","Genus","Abundance")], id=c("Site_name","Month","Abundance"))
cor_genera_geochem_sulfurodm1 = cor_genera_geochem_sulfurodm1[, c("Site_name", "Month","value","Abundance")]
colnames(cor_genera_geochem_sulfurodm1) <- c("Site_name", "Month", "Genus", "Abundance")

cor_genera_geochem_sulfurodm2 <- melt(cor_genera_geochem_sulfuro[,c("Site_name","Month","Temp","DO","pH","Na","Li",
                                                            "Ba","B","As","Co","Cu","Se")], id=c("Site_name","Month"))
colnames(cor_genera_geochem_sulfurodm2) <- c("Site_name", "Month", "variable", "value")

cor_genera_geochem_sulfurodm1dm2_melt <- merge(cor_genera_geochem_sulfurodm1, cor_genera_geochem_sulfurodm2, allow.cartesian=TRUE)

cor_genera_geochem_sulfurodm1dm2_melt_plot=ggplot(cor_genera_geochem_sulfurodm1dm2_melt) +
  geom_jitter(aes(Abundance,value, 
                  colour=variable)) + 
  geom_smooth(aes(Abundance,value, colour=variable), 
              method=lm, se=FALSE) +
  facet_wrap(~variable, scales="free_x") +
  stat_cor(aes(Abundance,value,
               color = variable), 
           label.y = 370, label.x = 0.3)+
  stat_poly_eq(aes(Abundance,value,
        color = variable, label = ..eq.label..),
    formula = y ~ x, 
    label.y = 300, label.x = 0.3,
    parse = TRUE)
```

```{r echo=FALSE, message=FALSE, warning=FALSE, fig.width=16,fig.height=16}
cor_genera_geochem_galdm1dm2_melt_plot + ggtitle("Gallionella")
cor_genera_geochem_siddm1dm2_melt_plot + ggtitle("Sideroxydans")
cor_genera_geochem_ferdm1dm2_melt_plot + ggtitle("Ferriphaselus")
cor_genera_geochem_sulfuridm1dm2_melt_plot + ggtitle("Sulfuricurvum")
cor_genera_geochem_sulfurodm1dm2_melt_plot + ggtitle("Sulfurovum")
```










